diff -ruN fusionforge-4.8.3-original/common/docman/Parsedata.class.php fusionforge-4.8.3-ereg_preg/common/docman/Parsedata.class.php
--- fusionforge-4.8.3-original/common/docman/Parsedata.class.php	2009-06-07 10:49:31.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/common/docman/Parsedata.class.php	2010-05-06 04:49:35.260351176 -0700
@@ -78,7 +78,7 @@
 		$cmd = $this->p_path.$this->parsers["text/plain"];
 		$cmd = "$cmd $filename";
 		$rep1 = shell_exec ($cmd);
-		return ereg_replace ("\n", " ", "$rep $rep1");
+		return preg_replace("/\n/", " ", "$rep $rep1");
 	}
 	
 	
diff -ruN fusionforge-4.8.3-original/common/frs/FRSPackage.class.php fusionforge-4.8.3-ereg_preg/common/frs/FRSPackage.class.php
--- fusionforge-4.8.3-original/common/frs/FRSPackage.class.php	2009-09-20 11:46:33.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/common/frs/FRSPackage.class.php	2010-05-06 04:52:44.433027838 -0700
@@ -239,7 +239,7 @@
 	 *  @return string  The name of this package.
 	 */
 	function getFileName() {
-		return eregi_replace("[^-A-Z0-9_\.]",'',$this->data_array['name']);
+		return preg_replace("/[^-A-Z0-9_\.]/i",'',$this->data_array['name']);
 	}
 
 	/**
diff -ruN fusionforge-4.8.3-original/common/frs/FRSRelease.class.php fusionforge-4.8.3-ereg_preg/common/frs/FRSRelease.class.php
--- fusionforge-4.8.3-original/common/frs/FRSRelease.class.php	2009-06-07 10:49:32.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/common/frs/FRSRelease.class.php	2010-05-06 04:53:11.773415277 -0700
@@ -226,7 +226,7 @@
 	 *  @return string  The filename of this release.
 	 */
 	function getFileName() {
-		return eregi_replace("[^-A-Z0-9_\.]",'',$this->data_array['name']);
+		return preg_replace("/[^-A-Z0-9_\.]/i",'',$this->data_array['name']);
 	}
 
 	/**
diff -ruN fusionforge-4.8.3-original/common/include/account.php fusionforge-4.8.3-ereg_preg/common/include/account.php
--- fusionforge-4.8.3-original/common/include/account.php	2009-09-20 11:46:33.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/common/include/account.php	2010-05-06 05:11:16.838900496 -0700
@@ -61,15 +61,16 @@
 		return 0;
 	}
 
-	if (!ereg('^[a-z][-a-z0-9_]+$', $name)) {
+	if (!preg_match('/^[a-z][-a-z0-9_]+$/', $name)) {
 		$GLOBALS['register_error'] = _('Illegal character in name.');
 		return 0;
 	}
 
 	// illegal names
-	if (eregi("^((root)|(bin)|(daemon)|(adm)|(lp)|(sync)|(shutdown)|(halt)|(mail)|(news)"
-		. "|(uucp)|(operator)|(games)|(mysql)|(httpd)|(nobody)|(dummy)"
-		. "|(www)|(cvs)|(shell)|(ftp)|(irc)|(debian)|(ns)|(download))$",$name)) {
+	$regExpReservedNames = "^(root|bin|daemon|adm|lp|sync|shutdown|halt|mail|news|"
+		. "uucp|operator|games|mysql|httpd|nobody|dummy|www|cvs|shell|ftp|irc|"
+		. "debian|ns|download)$";
+	if( preg_match("/$regExpReservedNames/i", $name) ) {
 		$GLOBALS['register_error'] = _('Name is reserved.');
 		return 0;
 	}
@@ -83,7 +84,7 @@
 			return 0;
 		}
 	}
-	if (eregi("^(anoncvs_)",$name)) {
+	if (preg_match("/^(anoncvs_)/i",$name)) {
 		$GLOBALS['register_error'] = _('Name is reserved for CVS.');
 		return 0;
 	}
@@ -102,14 +103,16 @@
 	if (!account_namevalid($name)) return 0;
 	
 	// illegal names
-	if (eregi("^((www[0-9]?)|(cvs[0-9]?)|(shell[0-9]?)|(ftp[0-9]?)|(irc[0-9]?)|(news[0-9]?)"
-		. "|(mail[0-9]?)|(ns[0-9]?)|(download[0-9]?)|(pub)|(users)|(compile)|(lists)"
-		. "|(slayer)|(orbital)|(tokyojoe)|(webdev)|(projects)|(cvs)|(slayer)|(monitor)|(mirrors?))$",$name)) {
+	$regExpReservedGroupNames = "^(www[0-9]?|cvs[0-9]?|shell[0-9]?|ftp[0-9]?|"
+		. "irc[0-9]?|news[0-9]?|mail[0-9]?|ns[0-9]?|download[0-9]?|pub|users|"
+		. "compile|lists|slayer|orbital|tokyojoe|webdev|projects|cvs|monitor|"
+		. "mirrors?)$";
+	if(preg_match("/$regExpReservedGroupNames/i",$name)) {
 		$GLOBALS['register_error'] = _('Name is reserved for DNS purposes.');
 		return 0;
 	}
 
-	if (eregi("_",$name)) {
+	if(preg_match("/_/",$name)) {
 		$GLOBALS['register_error'] = _('Group name cannot contain underscore for DNS reasons.');
 		return 0;
 	}
@@ -215,7 +218,7 @@
 		if ($current == $this_shell) {
 			echo "<option selected value=$this_shell>$this_shell</option>\n";
 		} else {
-			if (! ereg("^#",$this_shell)){
+			if (!preg_match("/^#/",$this_shell)){
 				echo "<option value=$this_shell>$this_shell</option>\n";
 			}
 		}
diff -ruN fusionforge-4.8.3-original/common/include/MailParser.class.php fusionforge-4.8.3-ereg_preg/common/include/MailParser.class.php
--- fusionforge-4.8.3-original/common/include/MailParser.class.php	2009-06-07 10:49:32.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/common/include/MailParser.class.php	2010-05-06 05:11:54.229437549 -0700
@@ -64,7 +64,7 @@
 					//
 					//	See if line starts with tab, if so ignore it for now
 					//
-					if (!ereg('^[A-z]',$lines[$i])) {
+					if (!preg_match('/^[A-z]/',$lines[$i])) {
 						$header[$lastheader] = $header[$lastheader]."\n".$lines[$i];
 					} else {
 						$pos = (strpos($lines[$i],':'));
diff -ruN fusionforge-4.8.3-original/common/include/system/LDAP.class.php fusionforge-4.8.3-ereg_preg/common/include/system/LDAP.class.php
--- fusionforge-4.8.3-original/common/include/system/LDAP.class.php	2009-06-07 10:49:31.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/common/include/system/LDAP.class.php	2010-05-06 05:13:18.300645854 -0700
@@ -55,7 +55,7 @@
 			return '?';
 		}
 	
-		return ereg_replace("[\x80-\xff]","?",$str);
+		return preg_replace("/[\x80-\xff]/","?",$str);
 	}
 
 	/*
diff -ruN fusionforge-4.8.3-original/common/include/User.class.php fusionforge-4.8.3-ereg_preg/common/include/User.class.php
--- fusionforge-4.8.3-original/common/include/User.class.php	2009-09-20 11:46:33.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/common/include/User.class.php	2010-05-06 05:16:54.433756892 -0700
@@ -1071,7 +1071,7 @@
 	 *	@return	string	This user's SSH authorized (public) keys.
 	 */
 	function getAuthorizedKeys() {
-		return ereg_replace("###", "\n", $this->data_array['authorized_keys']);
+		return preg_replace("/###/", "\n", $this->data_array['authorized_keys']);
 	}
 
 	/**
@@ -1082,9 +1082,9 @@
 	 */
 	function setAuthorizedKeys($keys) {
 		$keys = trim($keys);
-		$keys = ereg_replace("\r\n", "\n", $keys); // Convert to Unix EOL
-		$keys = ereg_replace("\n+", "\n", $keys); // Remove empty lines
-		$keys = ereg_replace("\n", "###", $keys); // Convert EOL to marker
+		$keys = preg_replace("/\r\n/", "\n", $keys); // Convert to Unix EOL
+		$keys = preg_replace("/\n+/", "\n", $keys); // Remove empty lines
+		$keys = preg_replace("/\n/", "###", $keys); // Convert EOL to marker
 
 		$res = db_query_params ('UPDATE users SET authorized_keys=$1 WHERE user_id=$2',
 					array ($keys,
diff -ruN fusionforge-4.8.3-original/common/include/utils.php fusionforge-4.8.3-ereg_preg/common/include/utils.php
--- fusionforge-4.8.3-original/common/include/utils.php	2009-11-30 11:28:18.000000000 -0800
+++ fusionforge-4.8.3-ereg_preg/common/include/utils.php	2010-05-12 20:52:07.741780378 -0700
@@ -470,9 +470,14 @@
 		// angle brackets in the URL
 		// (RFC2396; http://www.w3.org/Addressing/URL/5.1_Wrappers.html)
 		$line = str_replace('&gt;', "\1", $line);
-		$line = eregi_replace("([ \t]|^)www\."," http://www.",$line);
-		$text = eregi_replace("([[:alnum:]]+)://([^[:space:]<\1]*)([[:alnum:]#?/&=])", "<a href=\"\\1://\\2\\3\" target=\"_new\">\\1://\\2\\3</a>", $line);
-		$text = eregi_replace("([[:space:]]|^)(([a-z0-9_]|\\-|\\.)+@([^[:space:]]*)([[:alnum:]-]))", "\\1<a href=\"mailto:\\2\" target=\"_new\">\\2</a>", $text);
+		$line = preg_replace( "/([ \t]|^)www\./i", " http://www.", $line);
+		$text = preg_replace( "/([[:alnum:]]+):\/\/([^[:space:]<\1]*)([[:alnum:]#?\/&=])/i", 
+			"<a href=\"\\1://\\2\\3\" target=\"_new\">\\1://\\2\\3</a>", $line);
+		$text = preg_replace( 
+			"/([[:space:]]|^)(([a-z0-9_]|\\-|\\.)+@([^[:space:]]*)([[:alnum:]-]))/i", 
+			"\\1<a href=\"mailto:\\2\" target=\"_new\">\\2</a>", 
+			$text
+			);
 		$text = str_replace("\1", '&gt;', $text);
 		$newText .= $text;
 	}
@@ -755,7 +760,9 @@
  *
  */
 function validate_email ($address) {
-	return (ereg('^[-!#$%&\'*+\\./0-9=?A-Z^_`a-z{|}~]+'. '@'. '[-!#$%&\'*+\\/0-9=?A-Z^_`a-z{|}~]+\.' . '[-!#$%&\'*+\\./0-9=?A-Z^_`a-z{|}~]+$', $address) !== false);
+	return ( preg_match( 
+	    "/^[-!#$%&\'*+\\.\/0-9=?A-Z^_`a-z{|}~]+@[-!#$%&\'*+\\/0-9=?A-Z^_`a-z{|}~]+\.[-!#$%&\'*+\\.\/0-9=?A-Z^_`a-z{|}~]+$/", 
+	    $address) );
 }
 
 /**
@@ -791,7 +798,7 @@
  */
 function util_is_valid_filename ($file) {
 	//bad char test
-	$invalidchars = eregi_replace("[-A-Z0-9+_\.]","",$file);
+	$invalidchars = preg_replace("/[-A-Z0-9+_\.]/i","",$file);
 
 	if (!empty($invalidchars)) {
 		return false;
@@ -814,14 +821,16 @@
 function valid_hostname ($hostname = "xyz") {
 
 	//bad char test
-	$invalidchars = eregi_replace("[-A-Z0-9\.]","",$hostname);
+	$invalidchars = preg_replace("/[-A-Z0-9\.]/i","",$hostname);
 
 	if (!empty($invalidchars)) {
 		return false;
 	}
 
 	//double dot, starts with a . or -
-	if (ereg("\.\.",$hostname) || ereg("^\.",$hostname) || ereg("^\-",$hostname)) {
+	if ( preg_match("/\.\./",$hostname) || 
+	     preg_match("/^\./",$hostname)  || 
+	     preg_match("/^\-/",$hostname) ) {
 		return false;
 	}
 
@@ -944,7 +953,7 @@
 	$url .= $GLOBALS['sys_default_domain'] ;
 	$url .= $GLOBALS['sys_urlprefix'] ;
 	$url .= "/" ;
-	$path = ereg_replace ("^/", "", $path) ;
+	$path = preg_replace ("/^\//", "", $path) ;
 	$url .= $path ;
 	return $url ;
 }
diff -ruN fusionforge-4.8.3-original/cronjobs/forum_gateway.php fusionforge-4.8.3-ereg_preg/cronjobs/forum_gateway.php
--- fusionforge-4.8.3-original/cronjobs/forum_gateway.php	2009-06-07 10:48:56.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/cronjobs/forum_gateway.php	2010-05-06 05:24:45.070793885 -0700
@@ -155,7 +155,7 @@
 			$this->Subject = addslashes(substr($subj,$parent_end+3));
 		}
 */
-		if (ereg('(\[)([0-9]*)(\])',$subj,$arr)) {
+		if (preg_match('/(\[)([0-9]*)(\])/',$subj,$arr)) {
 			$this->Parent=$arr[2];
 			$parent_end=(strpos($subj,'['.$arr[2].']')) + strlen('['.$arr[2].']');
 			$this->Subject = addslashes(substr($subj,$parent_end));
diff -ruN fusionforge-4.8.3-original/cronjobs/tracker_gateway.php fusionforge-4.8.3-ereg_preg/cronjobs/tracker_gateway.php
--- fusionforge-4.8.3-original/cronjobs/tracker_gateway.php	2009-06-07 10:48:56.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/cronjobs/tracker_gateway.php	2010-05-06 05:25:18.481303159 -0700
@@ -129,7 +129,7 @@
 		//we parse that ID to get the artifact that this should post to
 		//
 		$subj = $mp->getSubject();
-		if (ereg('(\[)([0-9]*)(\])',$subj,$arr)) {
+		if (preg_match('/(\[)([0-9]*)(\])/',$subj,$arr)) {
 		        $this->ArtifactId=$arr[2];
 			$artifactid_end=(strpos($subj,'['.$arr[2].']')) + strlen('['.$arr[2].']');
 			$this->Subject = addslashes(substr($subj,$artifactid_end));
diff -ruN fusionforge-4.8.3-original/plugins/fckeditor/www/editor/filemanager/browser/default/connectors/php/connector.php fusionforge-4.8.3-ereg_preg/plugins/fckeditor/www/editor/filemanager/browser/default/connectors/php/connector.php
--- fusionforge-4.8.3-original/plugins/fckeditor/www/editor/filemanager/browser/default/connectors/php/connector.php	2009-06-07 10:49:31.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/plugins/fckeditor/www/editor/filemanager/browser/default/connectors/php/connector.php	2010-05-06 05:38:46.383427134 -0700
@@ -37,14 +37,14 @@
 else
 	$GLOBALS["UserFilesPath"] = '/UserFiles/' ;
 
-if ( ! ereg( '/$', $GLOBALS["UserFilesPath"] ) )
+if ( ! preg_match( '/\/$/', $GLOBALS["UserFilesPath"] ) )
 	$GLOBALS["UserFilesPath"] .= '/' ;
 
 if ( strlen( $Config['UserFilesAbsolutePath'] ) > 0 ) 
 {
 	$GLOBALS["UserFilesDirectory"] = $Config['UserFilesAbsolutePath'] ;
 
-	if ( ! ereg( '/$', $GLOBALS["UserFilesDirectory"] ) )
+	if ( ! preg_match( '/\/$/', $GLOBALS["UserFilesDirectory"] ) )
 		$GLOBALS["UserFilesDirectory"] .= '/' ;
 }
 else
@@ -70,7 +70,7 @@
 		return ;
 
 	// Check the current folder syntax (must begin and start with a slash).
-	if ( ! ereg( '/$', $sCurrentFolder ) ) $sCurrentFolder .= '/' ;
+	if ( ! preg_match( '/\/$/', $sCurrentFolder ) ) $sCurrentFolder .= '/' ;
 	if ( strpos( $sCurrentFolder, '/' ) !== 0 ) $sCurrentFolder = '/' . $sCurrentFolder ;
 	
 	// Check for invalid folder paths (..)
diff -ruN fusionforge-4.8.3-original/plugins/fckeditor/www/editor/filemanager/browser/mcpuk/connectors/php/Commands/GetFoldersAndFiles.php fusionforge-4.8.3-ereg_preg/plugins/fckeditor/www/editor/filemanager/browser/mcpuk/connectors/php/Commands/GetFoldersAndFiles.php
--- fusionforge-4.8.3-original/plugins/fckeditor/www/editor/filemanager/browser/mcpuk/connectors/php/Commands/GetFoldersAndFiles.php	2009-06-07 10:49:30.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/plugins/fckeditor/www/editor/filemanager/browser/mcpuk/connectors/php/Commands/GetFoldersAndFiles.php	2010-05-06 05:41:32.615719394 -0700
@@ -75,7 +75,7 @@
 							//check if$fckphp_configured not to show this folder
 							$hide=false;
 							for($i=0;$i<sizeof($this->fckphp_config['ResourceAreas'][$this->type]['HideFolders']);$i++) 
-								$hide=(ereg($this->fckphp_config['ResourceAreas'][$this->type]['HideFolders'][$i],$filename)?true:$hide);
+								$hide=(preg_match("/$this->fckphp_config['ResourceAreas'][$this->type]['HideFolders'][$i]/",$filename)?true:$hide);
 							
 							if (!$hide) echo "\t\t<Folder name=\"$filename\" />\n";
 							
@@ -99,7 +99,7 @@
 					//check if$fckphp_configured not to show this file
 					$editable=$hide=false;
 					for($j=0;$j<sizeof($this->fckphp_config['ResourceAreas'][$this->type]['HideFiles']);$j++) 
-						$hide=(ereg($this->fckphp_config['ResourceAreas'][$this->type]['HideFiles'][$j],$files[$i])?true:$hide);
+						$hide=(preg_match("/$this->fckphp_config['ResourceAreas'][$this->type]['HideFiles'][$j]/",$files[$i])?true:$hide);
 					
 					if (!$hide) {
 						if ($this->fckphp_config['ResourceAreas'][$this->type]['AllowImageEditing']) 
diff -ruN fusionforge-4.8.3-original/plugins/fckeditor/www/editor/filemanager/browser/mcpuk/connectors/php/Commands/GetFolders.php fusionforge-4.8.3-ereg_preg/plugins/fckeditor/www/editor/filemanager/browser/mcpuk/connectors/php/Commands/GetFolders.php
--- fusionforge-4.8.3-original/plugins/fckeditor/www/editor/filemanager/browser/mcpuk/connectors/php/Commands/GetFolders.php	2009-06-07 10:49:30.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/plugins/fckeditor/www/editor/filemanager/browser/mcpuk/connectors/php/Commands/GetFolders.php	2010-05-06 05:47:07.170355828 -0700
@@ -48,7 +48,7 @@
 							//check if$fckphp_configured not to show this folder
 							$hide=false;
 							for($i=0;$i<sizeof($this->fckphp_config['ResourceAreas'][$this->type]['HideFolders']);$i++) 
-								$hide=(ereg($this->fckphp_config['ResourceAreas'][$this->type]['HideFolders'][$i],$filename)?true:$hide);
+								$hide=(preg_match("/$this->fckphp_config['ResourceAreas'][$this->type]['HideFolders'][$i]/",$filename)?true:$hide);
 							
 							if (!$hide) echo "<Folder name=\"$filename\" />\n";
 						}
diff -ruN fusionforge-4.8.3-original/plugins/mediawiki/usr/share/mediawiki/includes/GForgePre.php fusionforge-4.8.3-ereg_preg/plugins/mediawiki/usr/share/mediawiki/includes/GForgePre.php
--- fusionforge-4.8.3-original/plugins/mediawiki/usr/share/mediawiki/includes/GForgePre.php	2009-06-07 10:49:05.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/plugins/mediawiki/usr/share/mediawiki/includes/GForgePre.php	2010-05-06 05:47:45.830893510 -0700
@@ -137,7 +137,7 @@
         if (getStringFromServer('REQUEST_URI')!='/' && $expl_pathinfo[1]!='account' && $expl_pathinfo[1]!='export' ) exit_not_logged_in();
 	// Show proj* export even if not logged in when force login
 	// If not default web project page would be broken
-	if ($expl_pathinfo[1]=='export' && !ereg("^proj", $expl_pathinfo[2])) exit_not_logged_in();
+	if ($expl_pathinfo[1]=='export' && !preg_match("/^proj/", $expl_pathinfo[2])) exit_not_logged_in();
 }
 
 //insert this page view into the database
diff -ruN fusionforge-4.8.3-original/plugins/scmccase/common/CCasePlugin.class.php fusionforge-4.8.3-ereg_preg/plugins/scmccase/common/CCasePlugin.class.php
--- fusionforge-4.8.3-original/plugins/scmccase/common/CCasePlugin.class.php	2009-06-07 10:49:02.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/plugins/scmccase/common/CCasePlugin.class.php	2010-05-06 05:48:23.971424328 -0700
@@ -80,7 +80,7 @@
 		$project =& group_get_object($group_id);
 		
 		if ($project->usesPlugin ("scmccase")) {
-			$vob_tag = ereg_replace ("GROUPNAME", $project->getUnixName (), $this->tag_pattern) ;
+			$vob_tag = preg_replace("/GROUPNAME/", $project->getUnixName (), $this->tag_pattern) ;
 	
 			print '<h2>ClearCase</h2>
 		                <p>Documentation for ClearCase is probably available somewhere.
diff -ruN fusionforge-4.8.3-original/plugins/scmsvn/cronjobs/svn-stats.php fusionforge-4.8.3-ereg_preg/plugins/scmsvn/cronjobs/svn-stats.php
--- fusionforge-4.8.3-original/plugins/scmsvn/cronjobs/svn-stats.php	2009-06-07 10:49:27.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/plugins/scmsvn/cronjobs/svn-stats.php	2010-05-06 05:48:59.191914843 -0700
@@ -75,7 +75,7 @@
 			$end_time, $time_ok, $date_key, $user_list;
 	switch ($last_tag) {
 		case "AUTHOR":
-			$last_user = ereg_replace ('[^a-z0-9_-]', '', 
+			$last_user = preg_replace('/[^a-z0-9_-]/', '', 
 						   strtolower (trim ($chars))) ;
 			// We can save time by looking up users and caching them
 			if (!array_key_exists($last_user, $user_list)) {
diff -ruN fusionforge-4.8.3-original/plugins/webcalendar/www/includes/functions.php fusionforge-4.8.3-ereg_preg/plugins/webcalendar/www/includes/functions.php
--- fusionforge-4.8.3-original/plugins/webcalendar/www/includes/functions.php	2009-06-07 10:49:24.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/plugins/webcalendar/www/includes/functions.php	2010-05-06 05:44:33.338220213 -0700
@@ -472,11 +472,11 @@
  * @ignore
  */
 function get_web_browser () {
-  if ( ereg ( "MSIE [0-9]", getenv ( "HTTP_USER_AGENT" ) ) )
+  if ( preg_match( "/MSIE [0-9]/", getenv ( "HTTP_USER_AGENT" ) ) )
     return "MSIE";
-  if ( ereg ( "Mozilla/[234]", getenv ( "HTTP_USER_AGENT" ) ) )
+  if ( preg_match( "/Mozilla\/[234]/", getenv ( "HTTP_USER_AGENT" ) ) )
     return "Netscape";
-  if ( ereg ( "Mozilla/[5678]", getenv ( "HTTP_USER_AGENT" ) ) )
+  if ( preg_match( "/Mozilla\/[5678]/", getenv ( "HTTP_USER_AGENT" ) ) )
     return "Mozilla";
   return "Unknown";
 }
@@ -1026,7 +1026,7 @@
   } else {
     $langs = explode ( ",", $HTTP_ACCEPT_LANGUAGE );
     for ( $i = 0; $i < count ( $langs ); $i++ ) {
-     $l = strtolower ( trim ( ereg_replace(';.*', '', $langs[$i] ) ) );
+     $l = strtolower ( trim ( preg_replace('/;.*/', '', $langs[$i] ) ) );
       $ret .= "\"$l\" ";
       if ( ! empty ( $browser_languages[$l] ) ) {
         return $browser_languages[$l];
@@ -3413,10 +3413,10 @@
  *                (http or https)
  */
 function activate_urls ( $text ) {
-  $str = eregi_replace ( "(http://[^[:space:]$]+)",
-    "<a href=\"\\1\">\\1</a>", $text );
-  $str = eregi_replace ( "(https://[^[:space:]$]+)",
-    "<a href=\"\\1\">\\1</a>", $str );
+  $str = preg_replace( "/(http:\/\/[^[:space:]$]+)/i",
+	"<a href=\"\\1\">\\1</a>", $text );
+  $str = preg_replace( "/(https:\/\/[^[:space:]$]+)/i",
+	"<a href=\"\\1\">\\1</a>", $str );
   return $str;
 }
 
diff -ruN fusionforge-4.8.3-original/plugins/webcalendar/www/includes/translate.php fusionforge-4.8.3-ereg_preg/plugins/webcalendar/www/includes/translate.php
--- fusionforge-4.8.3-original/plugins/webcalendar/www/includes/translate.php	2009-06-07 10:49:24.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/plugins/webcalendar/www/includes/translate.php	2010-05-06 05:45:41.839170429 -0700
@@ -174,8 +174,8 @@
  */
 function tooltip ( $str ) {
   $ret = translate ( $str );
-  $ret = eregi_replace ( "<[^>]+>", "", $ret );
-  $ret = eregi_replace ( "\"", "'", $ret );
+  $ret = preg_replace( "/<[^>]+>/i", "", $ret );
+  $ret = preg_replace( "/\"/", "'", $ret );
   return $ret;
 }
 
diff -ruN fusionforge-4.8.3-original/plugins/wiki/www/lib/FileFinder.php fusionforge-4.8.3-ereg_preg/plugins/wiki/www/lib/FileFinder.php
--- fusionforge-4.8.3-original/plugins/wiki/www/lib/FileFinder.php	2009-06-07 10:49:22.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/plugins/wiki/www/lib/FileFinder.php	2010-05-06 05:52:13.264623168 -0700
@@ -156,8 +156,8 @@
      * @return bool True if path is absolute. 
      */
     function _is_abs($path) {
-        if (ereg('^/', $path)) return true;
-        elseif (isWindows() and (eregi('^[a-z]:[/\\]', $path))) return true;
+        if (preg_match('/^\//', $path)) return true;
+        elseif (isWindows() and (preg_match('%^[a-z]:[/\\]%i', $path))) return true;
         else return false;
     }
 
@@ -664,4 +664,4 @@
 // c-hanging-comment-ender-p: nil
 // indent-tabs-mode: nil
 // End:
-?>
\ No newline at end of file
+?>
diff -ruN fusionforge-4.8.3-original/plugins/wiki/www/lib/IniConfig.php fusionforge-4.8.3-ereg_preg/plugins/wiki/www/lib/IniConfig.php
--- fusionforge-4.8.3-original/plugins/wiki/www/lib/IniConfig.php	2009-06-07 10:49:22.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/plugins/wiki/www/lib/IniConfig.php	2010-05-06 05:52:45.845078725 -0700
@@ -897,7 +897,7 @@
                 define('USE_PATH_INFO', false);
                 break;
             default:
-                define('USE_PATH_INFO', ereg('\.(php3?|cgi)$', SCRIPT_NAME));
+                define('USE_PATH_INFO', preg_match('/\.(php3?|cgi)$/', SCRIPT_NAME));
                 break;
             }
         }
diff -ruN fusionforge-4.8.3-original/plugins/wiki/www/lib/loadsave.php fusionforge-4.8.3-ereg_preg/plugins/wiki/www/lib/loadsave.php
--- fusionforge-4.8.3-original/plugins/wiki/www/lib/loadsave.php	2009-06-07 10:49:22.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/plugins/wiki/www/lib/loadsave.php	2010-05-06 05:53:11.175433086 -0700
@@ -53,7 +53,7 @@
     $tmpl = Template('html', array('TITLE' => $title,
                                    'HEADER' => $title,
                                    'CONTENT' => $html ? $html : '%BODY%'));
-    echo ereg_replace('%BODY%.*', '', $tmpl->getExpansion($html));
+    echo preg_replace('/%BODY%.*/', '', $tmpl->getExpansion($html));
     $request->chunkOutput();
     
     // set marker for sendPageChangeNotification()
diff -ruN fusionforge-4.8.3-original/plugins/wiki/www/lib/nusoap/nusoap.php fusionforge-4.8.3-ereg_preg/plugins/wiki/www/lib/nusoap/nusoap.php
--- fusionforge-4.8.3-original/plugins/wiki/www/lib/nusoap/nusoap.php	2009-06-07 10:49:20.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/plugins/wiki/www/lib/nusoap/nusoap.php	2010-05-06 10:56:48.463896187 -0700
@@ -277,7 +277,7 @@
                     break;
                 }
             }
-            if($valueType=='arraySimple' || ereg('^ArrayOf',$type)){
+            if($valueType=='arraySimple' || preg_match('/^ArrayOf/',$type)){
                 $i = 0;
                 if(is_array($val) && count($val)> 0){
                     foreach($val as $v){
@@ -441,7 +441,7 @@
 function timestamp_to_iso8601($timestamp,$utc=true){
     $datestr = date('Y-m-d\TH:i:sO',$timestamp);
     if($utc){
-        $eregStr =
+        $pregStr =
             '([0-9]{4})-'.	// centuries & years CCYY-
             '([0-9]{2})-'.	// months MM-
             '([0-9]{2})'.	// days DD
@@ -451,7 +451,7 @@
             '([0-9]{2})(\.[0-9]*)?'. // seconds ss.ss...
             '(Z|[+\-][0-9]{2}:?[0-9]{2})?'; // Z to indicate UTC, -/+HH:MM:SS.SS... for local tz's
         
-        if(ereg($eregStr,$datestr,$regs)){
+        if(preg_match("/$pregStr/",$datestr,$regs)){
             return sprintf('%04d-%02d-%02dT%02d:%02d:%02dZ',$regs[1],$regs[2],$regs[3],$regs[4],$regs[5],$regs[6]);
         }
         return false;
@@ -467,7 +467,7 @@
 * @access   public
 */
 function iso8601_to_timestamp($datestr){
-    $eregStr =
+    $pregStr =
 	'([0-9]{4})-'.	// centuries & years CCYY-
 	'([0-9]{2})-'.	// months MM-
 	'([0-9]{2})'.	// days DD
@@ -476,7 +476,7 @@
 	'([0-9]{2}):'.	// minutes mm:
 	'([0-9]{2})(\.[0-9]+)?'. // seconds ss.ss...
 	'(Z|[+\-][0-9]{2}:?[0-9]{2})?'; // Z to indicate UTC, -/+HH:MM:SS.SS... for local tz's
-    if(ereg($eregStr,$datestr,$regs)){
+    if(preg_match("/$pregStr/",$datestr,$regs)){
         // not utc
         if($regs[8] != 'Z'){
             $op = substr($regs[8],0,1);
@@ -708,7 +708,7 @@
             if(count($attrs) > 0){
                 foreach($attrs as $k => $v){
                     // if ns declarations, add to class level array of valid namespaces
-                    if(ereg("^xmlns",$k)){
+                    if(preg_match("/^xmlns/",$k)){
                 	//$this->xdebug("$k: $v");
                 	//$this->xdebug('ns_prefix: '.$this->getPrefix($k));
                 	if($ns_prefix = substr(strrchr($k,':'),1)){
@@ -783,7 +783,7 @@
                     $this->currentComplexType = $attrs['name'];
                     $this->complexTypes[$this->currentComplexType] = $attrs;
                     $this->complexTypes[$this->currentComplexType]['typeClass'] = 'complexType';
-                    if(isset($attrs['base']) && ereg(':Array$',$attrs['base'])){
+                    if(isset($attrs['base']) && preg_match('/:Array$/',$attrs['base'])){
                         $this->complexTypes[$this->currentComplexType]['phpType'] = 'array';
                     } else {
                         $this->complexTypes[$this->currentComplexType]['phpType'] = 'struct';
@@ -952,7 +952,7 @@
 	*/
 	function expandQname($qname){
             // get element prefix
-            if(strpos($qname,':') && !ereg('^http://',$qname)){
+            if(strpos($qname,':') && !preg_match('%^http://%',$qname)){
                 // get unqualified name
                 $name = substr(strstr($qname,':'),1);
                 // get ns prefix
@@ -1430,7 +1430,7 @@
         $data = $this->incoming_payload."\r\n\r\n\r\n\r\n";
 		
         // remove 100 header
-        if(ereg('^HTTP/1.1 100',$data)){
+        if(preg_match('%^HTTP/1.1 100%',$data)){
             if($pos = strpos($data,"\r\n\r\n") ){
                 $data = ltrim(substr($data,$pos));
             } elseif($pos = strpos($data,"\n\n") ){
@@ -1604,7 +1604,7 @@
         //$t->setMarker('closed curl');
 		
         // remove 100 header
-        if(ereg('^HTTP/1.1 100',$data)){
+        if(preg_match('%^HTTP/1.1 100%',$data)){
             if($pos = strpos($data,"\r\n\r\n") ){
                 $data = ltrim(substr($data,$pos));
             } elseif($pos = strpos($data,"\n\n") ){
@@ -1838,7 +1838,7 @@
             $qs = $QUERY_STRING;
         }
         // gen wsdl
-        if(isset($qs) && ereg('wsdl', $qs) ){
+        if(isset($qs) && preg_match('/wsdl/', $qs) ){
             if($this->externalWSDLURL){
                 header('Location: '.$this->externalWSDLURL);
                 exit();
@@ -1908,7 +1908,7 @@
             // get the character encoding of the incoming request
             if(strpos($this->headers['Content-Type'],'=')){
                 $enc = str_replace('"','',substr(strstr($this->headers["Content-Type"],'='),1));
-                if(eregi('^(ISO-8859-1|US-ASCII|UTF-8)$',$enc)){
+                if(preg_match('/^(ISO-8859-1|US-ASCII|UTF-8)$/i',$enc)){
                     $this->xml_encoding = $enc;
                 } else {
                     $this->xml_encoding = 'us-ascii';
@@ -2527,7 +2527,7 @@
      */
     function start_element($parser, $name, $attrs)
     {
-        if ($this->status == 'schema' || ereg('schema$', $name)) {
+        if ($this->status == 'schema' || preg_match('/schema$/', $name)) {
             // $this->debug("startElement for $name ($attrs[name]). status = $this->status (".$this->getLocalPart($name).")");
             $this->status = 'schema';
             $this->schemaStartElement($parser, $name, $attrs);
@@ -2539,7 +2539,7 @@
             $this->depth_array[$depth] = $pos;
             $this->message[$pos] = array('cdata' => ''); 
             // get element prefix
-            if (ereg(':', $name)) {
+            if (preg_match('/:/', $name)) {
                 // get ns prefix
                 $prefix = substr($name, 0, strpos($name, ':')); 
                 // get ns
@@ -2551,7 +2551,7 @@
             if (count($attrs) > 0) {
                 foreach($attrs as $k => $v) {
                     // if ns declarations, add to class level array of valid namespaces
-                    if (ereg("^xmlns", $k)) {
+                    if (preg_match("/^xmlns/", $k)) {
                         if ($ns_prefix = substr(strrchr($k, ':'), 1)) {
                             $this->namespaces[$ns_prefix] = $v;
                         } else {
@@ -2720,7 +2720,7 @@
      */
     function end_element($parser, $name){ 
         // unset schema status
-        if (ereg('types$', $name) || ereg('schema$', $name)) {
+        if (preg_match('/types$/', $name) || preg_match('/schema$/', $name)) {
             $this->status = "";
         } 
         if ($this->status == 'schema') {
@@ -3344,7 +3344,7 @@
 			$key_localpart = $this->getLocalPart($key);
 			// if ns declarations, add to class level array of valid namespaces
             if($key_prefix == 'xmlns'){
-				if(ereg('^http://www.w3.org/[0-9]{4}/XMLSchema$',$value)){
+				if(preg_match('#^http://www.w3.org/[0-9]{4}/XMLSchema$#',$value)){
 					$this->XMLSchemaVersion = $value;
 					$this->namespaces['xsd'] = $this->XMLSchemaVersion;
 					$this->namespaces['xsi'] = $this->XMLSchemaVersion.'-instance';
@@ -3368,7 +3368,7 @@
 				// should do something here with the namespace of specified type?
 			} elseif($key_localpart == 'arrayType'){
 				$this->message[$pos]['type'] = 'array';
-				/* do arrayType ereg here
+				/* do arrayType preg here
 				[1]    arrayTypeValue    ::=    atype asize
 				[2]    atype    ::=    QName rank*
 				[3]    rank    ::=    '[' (',')* ']'
@@ -3377,7 +3377,7 @@
 				[6]    nextDimension    ::=    Digit+ ','
 				*/
 				$expr = '([A-Za-z0-9_]+):([A-Za-z]+[A-Za-z0-9_]+)\[([0-9]+),?([0-9]*)\]';
-				if(ereg($expr,$value,$regs)){
+				if(preg_match("/$expr/",$value,$regs)){
 					$this->message[$pos]['typePrefix'] = $regs[1];
 					$this->message[$pos]['arraySize'] = $regs[3];
 					$this->message[$pos]['arrayCols'] = $regs[4];
@@ -3847,7 +3847,7 @@
 		// detect transport
 		switch(true){
 			// http(s)
-			case ereg('^http',$this->endpoint):
+			case preg_match('/^http/',$this->endpoint):
 				$this->debug('transporting via HTTP');
 				if($this->persistentConnection && is_object($this->persistentConnection)){
 					$http =& $this->persistentConnection;
@@ -3867,10 +3867,10 @@
 					$http->setEncoding($this->http_encoding);
 				}
 				$this->debug('sending message, length: '.strlen($msg));
-				if(ereg('^http:',$this->endpoint)){
+				if(preg_match('/^http:/',$this->endpoint)){
 				//if(strpos($this->endpoint,'http:')){
 					$response = $http->send($msg,$timeout);
-				} elseif(ereg('^https',$this->endpoint)){
+				} elseif(preg_match('/^https:/',$this->endpoint)){
 				//} elseif(strpos($this->endpoint,'https:')){
 					//if(phpversion() == '4.3.0-dev'){
 						//$response = $http->send($msg,$timeout);
diff -ruN fusionforge-4.8.3-original/plugins/wiki/www/lib/pear/DB/ifx.php fusionforge-4.8.3-ereg_preg/plugins/wiki/www/lib/pear/DB/ifx.php
--- fusionforge-4.8.3-original/plugins/wiki/www/lib/pear/DB/ifx.php	2009-06-07 10:49:18.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/plugins/wiki/www/lib/pear/DB/ifx.php	2010-05-06 11:09:26.545037337 -0700
@@ -416,7 +416,7 @@
      */
     function errorCode($nativecode)
     {
-        if (ereg('SQLCODE=(.*)]', $nativecode, $match)) {
+        if (preg_match('/SQLCODE=(.*)]/', $nativecode, $match)) {
             $code = $match[1];
             if (isset($this->errorcode_map[$code])) {
                 return $this->errorcode_map[$code];
diff -ruN fusionforge-4.8.3-original/plugins/wiki/www/lib/pear/DB/ldap.php fusionforge-4.8.3-ereg_preg/plugins/wiki/www/lib/pear/DB/ldap.php
--- fusionforge-4.8.3-original/plugins/wiki/www/lib/pear/DB/ldap.php	2009-06-07 10:49:18.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/plugins/wiki/www/lib/pear/DB/ldap.php	2010-05-06 11:29:02.347949859 -0700
@@ -867,7 +867,7 @@
     function createSequence($seq_name)
     {
         // Extract $seq_id from DN
-        ereg("^([^,]*),", $seq_name, $regs);
+        preg_match("/^([^,]*),/", $seq_name, $regs);
         $seq_id = $regs[1];
 
         // Create the sequence entry
diff -ruN fusionforge-4.8.3-original/plugins/wiki/www/lib/plugin/HtmlConverter.php fusionforge-4.8.3-ereg_preg/plugins/wiki/www/lib/plugin/HtmlConverter.php
--- fusionforge-4.8.3-original/plugins/wiki/www/lib/plugin/HtmlConverter.php	2009-06-07 10:49:20.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/plugins/wiki/www/lib/plugin/HtmlConverter.php	2010-05-06 14:21:34.766151630 -0700
@@ -98,10 +98,15 @@
 
 	function _processA(&$file) {
 
-	    $file = eregi_replace(
-	    "<a([[:space:]]+)href([[:space:]]*)=([[:space:]]*)\"([-/.a-zA-Z0-9_~#@%$?&=:\200-\377\(\)[:space:]]+)\"([^>]*)>", "{{\\4}}", $file);
-	
-		$file = eregi_replace("{{([-/a-zA-Z0-9._~#@%$?&=:\200-\377\(\)[:space:]]+)}}([^<]+)</a>", "[ \\2 | \\1 ]", $file);
+	    // $file = eregi_replace(
+	    // "<a([[:space:]]+)href([[:space:]]*)=([[:space:]]*)\"([-/.a-zA-Z0-9_~#@%$?&=:\200-\377\(\)[:space:]]+)\"([^>]*)>", "{{\\4}}", $file);
+
+	    $pRegEx = "<a(\s+)href(\s*)=(\s*)\"([-/\.a-zA-Z0-9_~#@%$?&=:\200-\377\(\)\s]+)\"([^>]*)>";
+	    $file = preg_replace( "|$pRegEx|i", "{{\\4}}", $file );
+
+	    // $file = eregi_replace("{{([-/a-zA-Z0-9._~#@%$?&=:\200-\377\(\)[:space:]]+)}}([^<]+)</a>", "[ \\2 | \\1 ]", $file);
+	    $pRegEx = "{{([-/a-zA-Z0-9\._~#@%$?&=:\200-\377\(\)\s]+)}}([^<]+)</a>";
+	    $file = preg_replace( "|$pRegEx|i", "[ \\2 | \\1 ]", $file );
 	}
 
 	function _processIMG(&$file) {
@@ -145,29 +150,41 @@
 		
 		$file = str_replace ("\r\n", "\n", $file);
 		
-		$file = eregi_replace ("<h1[[:space:]]?[^>]*>", "\n\n!!!!", $file);
+		// $file = eregi_replace ("<h1[[:space:]]?[^>]*>", "\n\n!!!!", $file);
+		$file = preg_replace("|<h1[\s]?[^>]*>|i", "\n\n!!!!", $file);
 		
-		$file = eregi_replace ("<h2[[:space:]]?[^>]*>", "\n\n!!!", $file);
+		// $file = eregi_replace ("<h2[[:space:]]?[^>]*>", "\n\n!!!", $file);
+		$file = preg_replace("|<h2[\s]?[^>]*>|i", "\n\n!!!", $file);
 		
-		$file = eregi_replace ("<h3[[:space:]]?[^>]*>", "\n\n!!", $file);
+		// $file = eregi_replace ("<h3[[:space:]]?[^>]*>", "\n\n!!", $file);
+		$file = preg_replace("|<h3[\s]?[^>]*>|i", "\n\n!!", $file);
 		
-		$file = eregi_replace ("<h4[[:space:]]?[^>]*>", "\n\n!", $file);
+		// $file = eregi_replace ("<h4[[:space:]]?[^>]*>", "\n\n!", $file);
+		$file = preg_replace("|<h4[\s]?[^>]*>|i", "\n\n!", $file);
 		
-		$file = eregi_replace ("<h5[[:space:]]?[^>]*>", "\n\n__", $file);
+		// $file = eregi_replace ("<h5[[:space:]]?[^>]*>", "\n\n__", $file);
+		$file = preg_replace("|<h5[\s]?[^>]*>|i", "\n\n__", $file);
 		
-		$file = eregi_replace ("</h1>", "\n\n", $file);
+		// $file = eregi_replace ("</h1>", "\n\n", $file);
+		$file = preg_replace("|</h1>|i", "\n\n", $file);
 		
-		$file = eregi_replace ("</h2>", "\n\n", $file);
+		// $file = eregi_replace ("</h2>", "\n\n", $file);
+		$file = preg_replace("|</h2>|i", "\n\n", $file);
 		
-		$file = eregi_replace ("</h3>", "\n\n", $file);
+		// $file = eregi_replace ("</h3>", "\n\n", $file);
+		$file = preg_replace("|</h3>|i", "\n\n", $file);
 		
-		$file = eregi_replace ("</h4>", "\n\n", $file);
+		// $file = eregi_replace ("</h4>", "\n\n", $file);
+		$file = preg_replace("|</h4>|i", "\n\n", $file);
 		
-		$file = eregi_replace ("</h5>", "__\n\n", $file);
+		// $file = eregi_replace ("</h5>", "__\n\n", $file);
+		$file = preg_replace("|</h5>|i", "__\n\n", $file);
 		
-		$file = eregi_replace ("<hr[[:space:]]?[^>]*>", "\n----\n", $file);
+		// $file = eregi_replace ("<hr[[:space:]]?[^>]*>", "\n----\n", $file);
+		$file = preg_replace("|<hr[\s]?[^>]*>|i", "\n----\n", $file);
 	
-		$file = eregi_replace ("<li[[:space:]]?[^>]*>", "* ", $file);
+		// $file = eregi_replace ("<li[[:space:]]?[^>]*>", "* ", $file);
+		$file = preg_replace("|<li[\s]?[^>]*>|i", "* ", $file);
 
 		// strip all tags, except for <pre>, which is supported by wiki
 		// and <p>'s which will be converted after compression.		
diff -ruN fusionforge-4.8.3-original/plugins/wiki/www/lib/Request.php fusionforge-4.8.3-ereg_preg/plugins/wiki/www/lib/Request.php
--- fusionforge-4.8.3-original/plugins/wiki/www/lib/Request.php	2009-06-07 10:49:22.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/plugins/wiki/www/lib/Request.php	2010-05-06 14:22:48.337132224 -0700
@@ -739,7 +739,7 @@
                 }
                 $tmp_file .= '/' . basename($fileinfo['tmp_name']);
                 /* ending slash in php.ini upload_tmp_dir is required. */
-                if (realpath(ereg_replace('/+', '/', $tmp_file)) != realpath($fileinfo['tmp_name'])) {
+                if (realpath(preg_replace('|/+|', '/', $tmp_file)) != realpath($fileinfo['tmp_name'])) {
                     trigger_error(sprintf("Uploaded tmpfile illegal: %s != %s.",$tmp_file, $fileinfo['tmp_name']).
                     	          "\n".
                     	          "Probably illegal TEMP environment or upload_tmp_dir setting. ".
diff -ruN fusionforge-4.8.3-original/plugins/wiki/www/lib/WikiDB/adodb/drivers/adodb-sybase.inc.php fusionforge-4.8.3-ereg_preg/plugins/wiki/www/lib/WikiDB/adodb/drivers/adodb-sybase.inc.php
--- fusionforge-4.8.3-original/plugins/wiki/www/lib/WikiDB/adodb/drivers/adodb-sybase.inc.php	2009-06-07 10:49:22.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/plugins/wiki/www/lib/WikiDB/adodb/drivers/adodb-sybase.inc.php	2010-05-06 14:26:39.560344264 -0700
@@ -363,7 +363,7 @@
 	global $ADODB_sybase_mths;
 	
 		//Dec 30 2000 12:00AM
-		if (!ereg( "([A-Za-z]{3})[-/\. ]+([0-9]{1,2})[-/\. ]+([0-9]{4})"
+		if (!preg_match( "|([A-Za-z]{3})[-/\. ]+([0-9]{1,2})[-/\. ]+([0-9]{4})|"
 			,$v, $rr)) return parent::UnixDate($v);
 			
 		if ($rr[3] <= TIMESTAMP_FIRST_YEAR) return 0;
@@ -380,7 +380,7 @@
 	global $ADODB_sybase_mths;
 		//11.02.2001 Toni Tunkkari toni.tunkkari@finebyte.com
 		//Changed [0-9] to [0-9 ] in day conversion
-		if (!ereg( "([A-Za-z]{3})[-/\. ]([0-9 ]{1,2})[-/\. ]([0-9]{4}) +([0-9]{1,2}):([0-9]{1,2}) *([apAP]{0,1})"
+		if (!preg_match( "|([A-Za-z]{3})[-/\. ]([0-9 ]{1,2})[-/\. ]([0-9]{4}) +([0-9]{1,2}):([0-9]{1,2}) *([apAP]{0,1})|"
 			,$v, $rr)) return parent::UnixTimeStamp($v);
 		if ($rr[3] <= TIMESTAMP_FIRST_YEAR) return 0;
 		
diff -ruN fusionforge-4.8.3-original/plugins/wiki/www/lib/WikiDB/backend/cvs.php fusionforge-4.8.3-ereg_preg/plugins/wiki/www/lib/WikiDB/backend/cvs.php
--- fusionforge-4.8.3-original/plugins/wiki/www/lib/WikiDB/backend/cvs.php	2009-06-07 10:49:21.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/plugins/wiki/www/lib/WikiDB/backend/cvs.php	2010-05-06 14:29:53.733668313 -0700
@@ -97,7 +97,7 @@
             $this->_mkdir( $this->_repository, 0775 );
     
             // assume that the repository is a local directory, prefix :local:
-            if ( !ereg( "^:local:", $this->_repository ) ) {
+            if ( !preg_match( "/^:local:/", $this->_repository ) ) {
                 $this->_repository = ":local:" . $this->_repository;
             }
             
@@ -288,7 +288,7 @@
 
         // shift log data until we get to the 'revision X.X' line
         // FIXME: ensure that we don't enter an endless loop here
-        while ( !ereg( "^revision 1.([0-9]+)$", $logdata[0], $revInfo ) ) {
+        while ( !preg_match( "/^revision 1.([0-9]+)$/", $logdata[0], $revInfo ) ) {
             array_shift( $logdata );
         }
 
@@ -563,7 +563,7 @@
 
     function _strip_leading_underscore( &$item ) 
     {
-        $item = ereg_replace( "^_", "", $item );
+        $item = preg_replace( "/^_/", "", $item );
     }
 
     /**
@@ -645,7 +645,7 @@
 
         $cmdOutput = implode( "\n", $cmdOutput );
         $revInfo = array();
-        ereg( "\nnew revision: 1[.]([0-9]+); previous revision: ", $cmdOutput,
+        preg_match( "/\nnew revision: 1[.]([0-9]+); previous revision: /", $cmdOutput,
               $revInfo );
 
         $this->_cvsDebug( "CP: revInfo 0: $revInfo[0]" );
@@ -654,7 +654,7 @@
             $this->_cvsDebug( "CP: got revision information" );
             return $revInfo[1];
         } else {
-            ereg( "\ninitial revision: 1[.]([0-9]+)", $cmdOutput, $revInfo );
+            preg_match( "/\ninitial revision: 1[.]([0-9]+)/", $cmdOutput, $revInfo );
             if ( isset( $revInfo[1] ) ) {
                 $this->_cvsDebug( "CP: is initial release" );
                 return 1;
@@ -909,7 +909,7 @@
      */
     function _strip_newlines( &$item, $key )
     {
-        $item = ereg_replace( "\n$", "", $item );
+        $item = preg_replace( "/\n$/", "", $item );
     }
 
 } /* End of WikiDB_backend_cvs class */
@@ -1008,7 +1008,7 @@
     {
         do {
             $pageName = Cvs_Backend_Array_Iterator::next();
-        } while ( !eregi( $this->_searchString, $pageName ) );
+        } while ( !preg_match( "|$this->_searchString|i", $pageName ) );
 
         return $pageName;
     }
diff -ruN fusionforge-4.8.3-original/plugins/wiki/www/lib/WikiUserNew.php fusionforge-4.8.3-ereg_preg/plugins/wiki/www/lib/WikiUserNew.php
--- fusionforge-4.8.3-original/plugins/wiki/www/lib/WikiUserNew.php	2009-06-07 10:49:22.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/plugins/wiki/www/lib/WikiUserNew.php	2010-05-06 14:31:02.954898677 -0700
@@ -1842,7 +1842,7 @@
     }
     $Connect = @fsockopen ( $ConnectAddress, 25 );
     if ($Connect) {
-        if (ereg("^220", $Out = fgets($Connect, 1024))) {
+        if (preg_match("/^220/", $Out = fgets($Connect, 1024))) {
             fputs ($Connect, "HELO $HTTP_HOST\r\n");
             $Out = fgets ( $Connect, 1024 );
             fputs ($Connect, "MAIL FROM: <".$email.">\r\n");
@@ -1851,12 +1851,12 @@
             $To = fgets ($Connect, 1024);
             fputs ($Connect, "QUIT\r\n");
             fclose($Connect);
-            if (!ereg ("^250", $From)) {
+            if (!preg_match("/^250/", $From)) {
                 $result[0]=false;
                 $result[1]="Server rejected address: ". $From;
                 return $result;
             }
-            if (!ereg ( "^250", $To )) {
+            if (!preg_match( "/^250/", $To )) {
                 $result[0]=false;
                 $result[1]="Server rejected address: ". $To;
                 return $result;
diff -ruN fusionforge-4.8.3-original/www/admin/admin_table.php fusionforge-4.8.3-ereg_preg/www/admin/admin_table.php
--- fusionforge-4.8.3-original/www/admin/admin_table.php	2009-06-07 10:48:51.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/www/admin/admin_table.php	2010-05-06 05:32:36.727976815 -0700
@@ -218,7 +218,7 @@
 			$sql .= "$var='". htmlspecialchars($val) ."', ";
 		}
 	}
-	$sql = ereg_replace(', $', ' ', $sql);
+	$sql = preg_replace('/, $/', ' ', $sql);
 	$sql .= "WHERE $primary_key=$id";
 
 	if (db_query($sql)) {
diff -ruN fusionforge-4.8.3-original/www/docman/include/vtemplate.class.php fusionforge-4.8.3-ereg_preg/www/docman/include/vtemplate.class.php
--- fusionforge-4.8.3-original/www/docman/include/vtemplate.class.php	2009-06-07 10:48:52.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/www/docman/include/vtemplate.class.php	2010-05-06 05:30:42.766242522 -0700
@@ -206,7 +206,7 @@
 // pour débogage
 function message ($text)
 {
-	$t = ereg_replace("\n", "", $text);
+	$t = preg_replace("/\n/", "", $text);
   if (defined("FICHIER_MESSAGES_DEBUG"))
 	{
 	  $fichier = FICHIER_MESSAGES_DEBUG;
diff -ruN fusionforge-4.8.3-original/www/docman/search.php fusionforge-4.8.3-ereg_preg/www/docman/search.php
--- fusionforge-4.8.3-original/www/docman/search.php	2009-09-20 11:46:26.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/www/docman/search.php	2010-05-06 05:31:51.757292535 -0700
@@ -238,13 +238,13 @@
 	
 	$rep = $text; 
 	$rep = utf8_decode($rep);
-	$rep = ereg_replace ("é", "e", $rep);
-	$rep = ereg_replace ("è", "e", $rep);
-	$rep = ereg_replace ("ê", "e", $rep);
-	$rep = ereg_replace ("à", "a", $rep);
-	$rep = ereg_replace ("ù", "u", $rep);
-	$rep = ereg_replace ("ç", "c", $rep);
-	$rep = ereg_replace ("é", "e", $rep);
+	$rep = preg_replace ("/é/", "e", $rep);
+	$rep = preg_replace ("/è/", "e", $rep);
+	$rep = preg_replace ("/ê/", "e", $rep);
+	$rep = preg_replace ("/à/", "a", $rep);
+	$rep = preg_replace ("/ù/", "u", $rep);
+	$rep = preg_replace ("/ç/", "c", $rep);
+	$rep = preg_replace ("/é/", "e", $rep);
 	$rep = strtolower ($rep);
 	return $rep;
 }
diff -ruN fusionforge-4.8.3-original/www/forum/include/ForumHTML.class.php fusionforge-4.8.3-ereg_preg/www/forum/include/ForumHTML.class.php
--- fusionforge-4.8.3-original/www/forum/include/ForumHTML.class.php	2009-06-07 10:48:51.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/www/forum/include/ForumHTML.class.php	2010-05-06 05:33:43.268989103 -0700
@@ -531,7 +531,7 @@
 	if ($this->Forum->userCanPost()) {
 		if ($subject) {
 			//if this is a followup, put a RE: before it if needed
-			if (!eregi('RE:',$subject,$test)) {
+			if (!preg_match('/RE:/i',$subject,$test)) {
 				$subject ='RE: '.$subject;
 			}
 		}
diff -ruN fusionforge-4.8.3-original/www/include/browser.php fusionforge-4.8.3-ereg_preg/www/include/browser.php
--- fusionforge-4.8.3-original/www/include/browser.php	2009-06-07 10:48:51.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/www/include/browser.php	2010-05-06 03:45:50.518734144 -0700
@@ -61,13 +61,13 @@
 */
 
 
-if (ereg( 'MSIE ([0-9].[0-9]{1,2})',getStringFromServer('HTTP_USER_AGENT'),$log_version)) {
+if (preg_match( '/MSIE ([0-9].[0-9]{1,2})/',getStringFromServer('HTTP_USER_AGENT'),$log_version)) {
 	$GLOBALS['BROWSER_VER']=$log_version[1];
 	$GLOBALS['BROWSER_AGENT']='IE';
-} elseif (ereg( 'Opera ([0-9].[0-9]{1,2})',getStringFromServer('HTTP_USER_AGENT'),$log_version)) {
+} elseif (preg_match( '/Opera ([0-9].[0-9]{1,2})/',getStringFromServer('HTTP_USER_AGENT'),$log_version)) {
 	$GLOBALS['BROWSER_VER']=$log_version[1];
 	$GLOBALS['BROWSER_AGENT']='OPERA';
-} elseif (ereg( 'Mozilla/([0-9].[0-9]{1,2})',getStringFromServer('HTTP_USER_AGENT'),$log_version)) {
+} elseif (preg_match( '/Mozilla\/([0-9].[0-9]{1,2})/',getStringFromServer('HTTP_USER_AGENT'),$log_version)) {
 	$GLOBALS['BROWSER_VER']=$log_version[1];
 	$GLOBALS['BROWSER_AGENT']='MOZILLA';
 } else {
diff -ruN fusionforge-4.8.3-original/www/include/filechecks.php fusionforge-4.8.3-ereg_preg/www/include/filechecks.php
--- fusionforge-4.8.3-original/www/include/filechecks.php	2009-06-07 10:48:51.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/www/include/filechecks.php	2010-05-06 03:52:55.604759001 -0700
@@ -27,7 +27,7 @@
 		return 0;
 	}
 
-	if (ereg('^\.',$filename)) {
+	if (preg_match('/^\./',$filename)) {
 		return 0;
 	}
 
@@ -73,82 +73,82 @@
 
 	$filename = chop($filename);
 
-	if (ereg(".diff$",$filename)) {
+	if (preg_match("/\.diff$/",$filename)) {
 		$filetype = "diff";
 	}
-	elseif (ereg(".diff.gz$",$filename)) {
+	elseif (preg_match("/\.diff.gz$/",$filename)) {
 		$filetype = "diff/gz";
 		filechecks_gz($filename);
 	}
-	elseif (ereg(".asc$",$filename)) {
+	elseif (preg_match("/\.asc$/",$filename)) {
 		$filetype = "asc";
 	}
-	elseif (ereg(".bin$",$filename)) {
+	elseif (preg_match("/\.bin$/",$filename)) {
 		$filetype = "bin";
 	}
-	elseif (ereg(".exe$",$filename)) {
+	elseif (preg_match("/\.exe$/",$filename)) {
 		$filetype = "exe";
 	}
-	elseif (ereg(".jar$",$filename)) {
+	elseif (preg_match("/\.jar$/",$filename)) {
 		$filetype = "jar";
 	}
-	elseif (ereg(".lsm$",$filename)) {
+	elseif (preg_match("/\.lsm$/",$filename)) {
 		$filetype = "lsm";
 	}
-	elseif (ereg(".pdb$",$filename)) {
+	elseif (preg_match("/\.pdb$/",$filename)) {
 		$filetype = "pilot";
 	}
-	elseif (ereg(".pl$",$filename)) {
+	elseif (preg_match("/\.pl$/",$filename)) {
 		$filetype = "perl";
 	}
-	elseif (ereg(".py$",$filename)) {
+	elseif (preg_match("/\.py$/",$filename)) {
 		$filetype = "python";
 	}
-	elseif (ereg(".prc$",$filename)) {
+	elseif (preg_match("/\.prc$/",$filename)) {
 		$filetype = "pilot";
 	}
-	elseif (ereg(".sig$",$filename)) {
+	elseif (preg_match("/\.sig$/",$filename)) {
 		$filetype = "sig";
 	}
-	elseif (ereg(".tar.bz2$",$filename)) {
+	elseif (preg_match("/\.tar.bz2$/",$filename)) {
 		$filetype = "tar/bz2";
 	}
-	elseif (ereg(".tar.gz$",$filename)) {
+	elseif (preg_match("/\.tar.gz$/",$filename)) {
 		$filetype = "tar/gz";
 		filechecks_targz($filename);
 	}
-	elseif (ereg(".tgz$",$filename)) {
+	elseif (preg_match("/\.tgz$/",$filename)) {
 		$filetype = "tgz";
 	}
-	elseif (ereg(".zip$",$filename)) {
+	elseif (preg_match("/\.zip$/",$filename)) {
 		$filetype = "zip";
 	}
-	elseif (ereg(".shar.gz$",$filename)) {
+	elseif (preg_match("/\.shar.gz$/",$filename)) {
 		$filetype = "shar/gz";
 	}
-	elseif (ereg(".bz2$",$filename)) {
+	elseif (preg_match("/\.bz2$/",$filename)) {
 		$filetype = "bz2";
 	}
-	elseif (ereg(".gz$",$filename)) {
+	elseif (preg_match("/\.gz$/",$filename)) {
 		$filetype = "gz";
 		filechecks_gz($filename);
 	}
-	elseif (ereg(".i386.rpm$",$filename)) {
+	elseif (preg_match("/\.i386.rpm$/",$filename)) {
 		$filetype = "i386 rpm";
 	}
-	elseif (ereg(".alpha.rpm$",$filename)) {
+	elseif (preg_match("/\.alpha.rpm$/",$filename)) {
 		$filetype = "alpha rpm";
 	}
-	elseif (ereg(".src.rpm$",$filename)) {
+	elseif (preg_match("/\.src.rpm$/",$filename)) {
 		$filetype = "src rpm";
 	}
-	elseif (ereg(".rpm$",$filename)) {
+	elseif (preg_match("/\.rpm$/",$filename)) {
 		$filetype = "rpm";
 	}
-	elseif (ereg(".deb$",$filename)) {
+	elseif (preg_match("/\.deb$/",$filename)) {
 		$filetype = "deb";
 	}
-	elseif (ereg("\.([a-zA-Z]+)$",$filename,$regs)) {
+	elseif (preg_match("/\.([a-zA-Z]+)$/",$filename,$regs)) {
 		$filetype = $regs[1];		
 	} 
 
diff -ruN fusionforge-4.8.3-original/www/include/HTML_Graphs.php fusionforge-4.8.3-ereg_preg/www/include/HTML_Graphs.php
--- fusionforge-4.8.3-original/www/include/HTML_Graphs.php	2009-06-07 10:48:51.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/www/include/HTML_Graphs.php	2010-05-06 05:28:39.284362472 -0700
@@ -474,7 +474,7 @@
 	echo ">";
 
         // Decide if the value in bar is a color code or image.
-        if (ereg("^#", $bars[$i]))
+        if (preg_match("/^#/", $bars[$i]))
            {
 ?>
 
@@ -611,13 +611,13 @@
 <?php
         // Set background to a color if it starts with # or
         // an image otherwise.
-        if (ereg("^#", $dbars[$i])) { print 'style="background-color:' . $dbars[$i] . '">'; }
+        if (preg_match("/^#/", $dbars[$i])) { print 'style="background-color:' . $dbars[$i] . '">'; }
         else { print 'background="' . $dbars[$i] . '">'; }
 ?>
            <nowrap>
 <?php
         // Decide if the value in bar is a color code or image.
-        if (ereg("^#", $bars[$i]))
+        if (preg_match("/^#/", $bars[$i]))
            {
 ?>
             <table align="left" cellpadding="0" cellspacing="0"
diff -ruN fusionforge-4.8.3-original/www/include/logger.php fusionforge-4.8.3-ereg_preg/www/include/logger.php
--- fusionforge-4.8.3-original/www/include/logger.php	2009-09-20 11:46:25.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/www/include/logger.php	2010-05-12 16:28:51.761957334 -0700
@@ -32,7 +32,9 @@
 	//
 	//
 	if ( $GLOBALS['sys_urlprefix'] != '' ) {
-		$pathwithoutprefix=ereg_replace($GLOBALS['sys_urlprefix'],'',getStringFromServer('REQUEST_URI'));
+		$pathwithoutprefix=preg_replace( 
+			'/$GLOBALS[\'sys_urlprefix\']/','',
+			getStringFromServer('REQUEST_URI') );
 	} else {
 		$pathwithoutprefix=getStringFromServer('REQUEST_URI');
 	}
diff -ruN fusionforge-4.8.3-original/www/include/pre.php fusionforge-4.8.3-ereg_preg/www/include/pre.php
--- fusionforge-4.8.3-original/www/include/pre.php	2009-11-30 11:28:15.000000000 -0800
+++ fusionforge-4.8.3-ereg_preg/www/include/pre.php	2010-05-06 03:29:50.485277078 -0700
@@ -117,10 +117,15 @@
 //mandatory login
 if (!session_loggedin() && $sys_force_login == 1 ) {
 	$expl_pathinfo = explode('/',getStringFromServer('REQUEST_URI'));
-        if (getStringFromServer('REQUEST_URI')!='/' && $expl_pathinfo[1]!='account' && $expl_pathinfo[1]!='export' ) exit_not_logged_in();
+        if ( getStringFromServer('REQUEST_URI')!='/' && 
+	     $expl_pathinfo[1]!='account' && 
+	     $expl_pathinfo[1]!='export' ) 
+		exit_not_logged_in();
 	// Show proj* export even if not logged in when force login
 	// If not default web project page would be broken
-	if ($expl_pathinfo[1]=='export' && !ereg("^proj", $expl_pathinfo[2])) exit_not_logged_in();
+	if ( $expl_pathinfo[1]=='export' && 
+	     !preg_match("/^proj/", $expl_pathinfo[2]) ) 
+		exit_not_logged_in();
 }
 
 //insert this page view into the database
diff -ruN fusionforge-4.8.3-original/www/news/submit.php fusionforge-4.8.3-ereg_preg/www/news/submit.php
--- fusionforge-4.8.3-original/www/news/submit.php	2009-06-07 10:48:49.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/www/news/submit.php	2010-05-06 05:29:49.995439181 -0700
@@ -65,7 +65,7 @@
 
 				db_begin();
 				$f=new Forum(group_get_object($sys_news_group));
-				if (!$f->create(ereg_replace('[^_\.0-9a-z-]','-', strtolower($summary)),$details,1,'',0,0)) {
+				if (!$f->create(preg_replace('/[^_\.0-9a-z-]/','-', strtolower($summary)),$details,1,'',0,0)) {
 					db_rollback();
 					exit_error('Error',$f->getErrorMessage());
 				}
diff -ruN fusionforge-4.8.3-original/www/project/admin/project_admin_utils.php fusionforge-4.8.3-ereg_preg/www/project/admin/project_admin_utils.php
--- fusionforge-4.8.3-original/www/project/admin/project_admin_utils.php	2009-06-07 10:48:48.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/www/project/admin/project_admin_utils.php	2010-05-06 05:36:14.811293587 -0700
@@ -171,7 +171,7 @@
 			<tr '. $GLOBALS['HTML']->boxGetAltRowStyle($i) .'><td>'.$field.'</td><td>';
 			
 			if (is_numeric(db_result($result, $i, 'old_value'))) {
-				if (ereg("user|User", $field)) {
+				if (preg_match("/[Uu]ser/", $field)) {
 					echo user_getname(db_result($result, $i, 'old_value'));
 				} else {
 					echo db_result($result, $i, 'old_value');
diff -ruN fusionforge-4.8.3-original/www/project/admin/roleedit.php fusionforge-4.8.3-ereg_preg/www/project/admin/roleedit.php
--- fusionforge-4.8.3-original/www/project/admin/roleedit.php	2009-06-07 10:48:48.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/www/project/admin/roleedit.php	2010-05-06 05:35:20.270464342 -0700
@@ -133,12 +133,12 @@
 $j = 0;
 $keys = array_keys($role->role_values);
 for ($i=0; $i<count($keys); $i++) {
-        if ((!$group->usesForum() && ereg("forum", $keys[$i])) ||
-                (!$group->usesTracker() && ereg("tracker", $keys[$i])) ||
-                (!$group->usesPM() && ereg("pm", $keys[$i])) ||
-                (!$group->usesFRS() && ereg("frs", $keys[$i])) ||
-                (!$group->usesSCM() && ereg("scm", $keys[$i])) ||
-                (!$group->usesDocman() && ereg("docman", $keys[$i]))) {
+        if ((!$group->usesForum() && preg_match("/forum/", $keys[$i])) ||
+                (!$group->usesTracker() && preg_match("/tracker/", $keys[$i])) ||
+                (!$group->usesPM() && preg_match("/pm/", $keys[$i])) ||
+                (!$group->usesFRS() && preg_match("/frs/", $keys[$i])) ||
+                (!$group->usesSCM() && preg_match("/scm/", $keys[$i])) ||
+                (!$group->usesDocman() && preg("/docman/", $keys[$i]))) {
 
                 //We don't display modules not used
 
diff -ruN fusionforge-4.8.3-original/www/scm/viewvc.php fusionforge-4.8.3-ereg_preg/www/scm/viewvc.php
--- fusionforge-4.8.3-original/www/scm/viewvc.php	2009-06-07 10:48:50.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/www/scm/viewvc.php	2010-05-06 04:42:47.604534565 -0700
@@ -37,7 +37,7 @@
 	}
 }
 // Remove eventual leading /root/ or root/
-$projectName = ereg_replace('^..[^/]*/','', $projectName);
+$projectName = preg_replace('%^..[^/]*/%','', $projectName);
 if (!$projectName) {
 	exit_no_group();
 }
diff -ruN fusionforge-4.8.3-original/www/sendmessage.php fusionforge-4.8.3-ereg_preg/www/sendmessage.php
--- fusionforge-4.8.3-original/www/sendmessage.php	2009-06-07 10:48:52.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/www/sendmessage.php	2010-05-13 00:33:05.987544961 -0700
@@ -44,7 +44,7 @@
 	}
 }
 
-if ($toaddress && !eregi($GLOBALS['sys_default_domain'],$toaddress)) {
+if ($toaddress && !preg_match('/$GLOBALS[\'sys_default_domain\']/i',$toaddress)) {
 	exit_error(_('Error'),sprintf(_('You can only send to addresses @<em>%1$s</em>.'),$GLOBALS['sys_default_domain']));
 }
 
@@ -77,7 +77,7 @@
 		/*
 			send it to the toaddress
 		*/
-		$to=eregi_replace('_maillink_','@',$toaddress);
+		$to=preg_replace('/_maillink_/i','@',$toaddress);
 		$to = util_remove_CRLF($to);
 		util_send_message($to,stripslashes($subject),stripslashes($body),$email,'',$name);
 		$HTML->header(array('title'=>$GLOBALS['sys_name'].' ' ._('Contact')   ));
diff -ruN fusionforge-4.8.3-original/www/soap/nusoap.php fusionforge-4.8.3-ereg_preg/www/soap/nusoap.php
--- fusionforge-4.8.3-original/www/soap/nusoap.php	2009-11-30 11:28:17.000000000 -0800
+++ fusionforge-4.8.3-ereg_preg/www/soap/nusoap.php	2010-05-06 04:23:29.987853912 -0700
@@ -562,7 +562,7 @@
 			case (is_array($val) || $type):
 				// detect if struct or array
 				$valueType = $this->isArraySimpleOrStruct($val);
-                if($valueType=='arraySimple' || ereg('^ArrayOf',$type)){
+                if($valueType=='arraySimple' || preg_match('/^ArrayOf/',$type)){
 			   		$this->debug("serialize_val: serialize array");
 					$i = 0;
 					if(is_array($val) && count($val)> 0){
@@ -765,7 +765,7 @@
 	*/
 	function expandQname($qname){
 		// get element prefix
-		if(strpos($qname,':') && !ereg('^http://',$qname)){
+		if(strpos($qname,':') && !preg_match('/^http:\/\//',$qname)){
 			// get unqualified name
 			$name = substr(strstr($qname,':'),1);
 			// get ns prefix
@@ -904,7 +904,7 @@
 function timestamp_to_iso8601($timestamp,$utc=true){
 	$datestr = date('Y-m-d\TH:i:sO',$timestamp);
 	if($utc){
-		$eregStr =
+		$pregStr =
 		'([0-9]{4})-'.	// centuries & years CCYY-
 		'([0-9]{2})-'.	// months MM-
 		'([0-9]{2})'.	// days DD
@@ -914,7 +914,7 @@
 		'([0-9]{2})(\.[0-9]*)?'. // seconds ss.ss...
 		'(Z|[+\-][0-9]{2}:?[0-9]{2})?'; // Z to indicate UTC, -/+HH:MM:SS.SS... for local tz's
 
-		if(ereg($eregStr,$datestr,$regs)){
+		if(preg_match("/$pregStr/",$datestr,$regs)){
 			return sprintf('%04d-%02d-%02dT%02d:%02d:%02dZ',$regs[1],$regs[2],$regs[3],$regs[4],$regs[5],$regs[6]);
 		}
 		return false;
@@ -930,7 +930,7 @@
 * @access   public
 */
 function iso8601_to_timestamp($datestr){
-	$eregStr =
+	$pregStr =
 	'([0-9]{4})-'.	// centuries & years CCYY-
 	'([0-9]{2})-'.	// months MM-
 	'([0-9]{2})'.	// days DD
@@ -939,7 +939,7 @@
 	'([0-9]{2}):'.	// minutes mm:
 	'([0-9]{2})(\.[0-9]+)?'. // seconds ss.ss...
 	'(Z|[+\-][0-9]{2}:?[0-9]{2})?'; // Z to indicate UTC, -/+HH:MM:SS.SS... for local tz's
-	if(ereg($eregStr,$datestr,$regs)){
+	if(preg_match("/$pregStr/",$datestr,$regs)){
 		// not utc
 		if($regs[8] != 'Z'){
 			$op = substr($regs[8],0,1);
@@ -1267,7 +1267,7 @@
         if(count($attrs) > 0){
         	foreach($attrs as $k => $v){
                 // if ns declarations, add to class level array of valid namespaces
-				if(ereg("^xmlns",$k)){
+				if(preg_match("/^xmlns/",$k)){
                 	//$this->xdebug("$k: $v");
                 	//$this->xdebug('ns_prefix: '.$this->getPrefix($k));
                 	if($ns_prefix = substr(strrchr($k,':'),1)){
@@ -1379,7 +1379,7 @@
 					//                        minOccurs="0" maxOccurs="unbounded" />
 					//                </sequence>
 					//            </complexType>
-					if(isset($attrs['base']) && ereg(':Array$',$attrs['base'])){
+					if(isset($attrs['base']) && preg_match('/:Array$/',$attrs['base'])){
 						$this->xdebug('complexType is unusual array');
 						$this->complexTypes[$this->currentComplexType]['phpType'] = 'array';
 					} else {
@@ -1399,7 +1399,7 @@
 					//                        minOccurs="0" maxOccurs="unbounded" />
 					//                </sequence>
 					//            </complexType>
-					if(isset($attrs['base']) && ereg(':Array$',$attrs['base'])){
+					if(isset($attrs['base']) && preg_match('/:Array$/',$attrs['base'])){
 						$this->xdebug('complexType is unusual array');
 						$this->complexTypes[$this->currentComplexType]['phpType'] = 'array';
 					} else {
@@ -1809,7 +1809,7 @@
 		} elseif(isset($this->attributes[$type])){
 			$this->xdebug("in getTypeDef, found attribute $type");
 			return $this->attributes[$type];
-		} elseif (ereg('_ContainedType$', $type)) {
+		} elseif (preg_match('/_ContainedType$/', $type)) {
 			$this->xdebug("in getTypeDef, have an untyped element $type");
 			$typeDef['typeClass'] = 'simpleType';
 			$typeDef['phpType'] = 'scalar';
@@ -2173,7 +2173,7 @@
 			$this->ch_options = $curl_options;
 		}
 		$this->use_curl = $use_curl;
-		ereg('\$Revisio' . 'n: ([^ ]+)', $this->revision, $rev);
+		preg_match('/\$Revisio/' . 'n: ([^ ]+)', $this->revision, $rev);
 		$this->setHeader('User-Agent', $this->title.'/'.$this->version.' ('.$rev[1].')');
 	}
 
@@ -2936,7 +2936,7 @@
 				}
 			}
 			// remove 100 headers
-			if (isset($lb) && ereg('^HTTP/1.1 100',$data)) {
+			if (isset($lb) && preg_match('/^HTTP\/1.1 100/',$data)) {
 				unset($lb);
 				$data = '';
 			}//
@@ -3102,7 +3102,7 @@
 		if ($data == '') {
 			// have nothing left; just remove 100 header(s)
 			$data = $savedata;
-			while (ereg('^HTTP/1.1 100',$data)) {
+			while (preg_match('/^HTTP\/1.1 100/',$data)) {
 				if ($pos = strpos($data,"\r\n\r\n")) {
 					$data = ltrim(substr($data,$pos));
 				} elseif($pos = strpos($data,"\n\n") ) {
@@ -3654,7 +3654,7 @@
 		}
 		$this->debug("In service, query string=$qs");
 
-		if (ereg('wsdl', $qs) ){
+		if (preg_match('/wsdl/', $qs) ){
 			$this->debug("In service, this is a request for WSDL");
 			if($this->externalWSDLURL){
               if (strpos($this->externalWSDLURL,"://")!==false) { // assume URL
@@ -3725,7 +3725,7 @@
 			// get the character encoding of the incoming request
 			if(isset($this->headers['content-type']) && strpos($this->headers['content-type'],'=')){
 				$enc = str_replace('"','',substr(strstr($this->headers["content-type"],'='),1));
-				if(eregi('^(ISO-8859-1|US-ASCII|UTF-8)$',$enc)){
+				if(preg_match('/^(ISO-8859-1|US-ASCII|UTF-8)$/i',$enc)){
 					$this->xml_encoding = strtoupper($enc);
 				} else {
 					$this->xml_encoding = 'US-ASCII';
@@ -3754,7 +3754,7 @@
 						$enc = substr(strstr($v, '='), 1);
 						$enc = str_replace('"', '', $enc);
 						$enc = str_replace('\\', '', $enc);
-						if (eregi('^(ISO-8859-1|US-ASCII|UTF-8)$', $enc)) {
+						if (preg_match('/^(ISO-8859-1|US-ASCII|UTF-8)$/i', $enc)) {
 							$this->xml_encoding = strtoupper($enc);
 						} else {
 							$this->xml_encoding = 'US-ASCII';
@@ -3788,7 +3788,7 @@
 						$enc = substr(strstr($v, '='), 1);
 						$enc = str_replace('"', '', $enc);
 						$enc = str_replace('\\', '', $enc);
-						if (eregi('^(ISO-8859-1|US-ASCII|UTF-8)$', $enc)) {
+						if (preg_match('/^(ISO-8859-1|US-ASCII|UTF-8)$/i', $enc)) {
 							$this->xml_encoding = strtoupper($enc);
 						} else {
 							$this->xml_encoding = 'US-ASCII';
@@ -4126,7 +4126,7 @@
         	$payload .= $this->getDebugAsXMLComment();
         }
 		$this->outgoing_headers[] = "Server: $this->title Server v$this->version";
-		ereg('\$Revisio' . 'n: ([^ ]+)', $this->revision, $rev);
+		preg_match('/\$Revision: ([^ ]+)/', $this->revision, $rev);
 		$this->outgoing_headers[] = "X-SOAP-Server: $this->title/$this->version (".$rev[1].")";
 		// Let the Web server decide about this
 		//$this->outgoing_headers[] = "Connection: Close\r\n";
@@ -4214,7 +4214,7 @@
 		if (strpos($headers['content-type'], '=')) {
 			$enc = str_replace('"', '', substr(strstr($headers["content-type"], '='), 1));
 			$this->debug('Got response encoding: ' . $enc);
-			if(eregi('^(ISO-8859-1|US-ASCII|UTF-8)$',$enc)){
+			if(preg_match('/^(ISO-8859-1|US-ASCII|UTF-8)$/i',$enc)){
 				$this->xml_encoding = strtoupper($enc);
 			} else {
 				$this->xml_encoding = 'US-ASCII';
@@ -4781,7 +4781,7 @@
             $this->currentSchema->schemaStartElement($parser, $name, $attrs);
             $this->appendDebug($this->currentSchema->getDebug());
             $this->currentSchema->clearDebug();
-        } elseif (ereg('schema$', $name)) {
+        } elseif (preg_match('/schema$/', $name)) {
         	$this->debug('Parsing WSDL schema');
             // $this->debug("startElement for $name ($attrs[name]). status = $this->status (".$this->getLocalPart($name).")");
             $this->status = 'schema';
@@ -4800,7 +4800,7 @@
             if (count($attrs) > 0) {
 				// register namespace declarations
                 foreach($attrs as $k => $v) {
-                    if (ereg("^xmlns", $k)) {
+                    if (preg_match("/^xmlns/", $k)) {
                         if ($ns_prefix = substr(strrchr($k, ':'), 1)) {
                             $this->namespaces[$ns_prefix] = $v;
                         } else {
@@ -4825,7 +4825,7 @@
                 $attrs = array();
             } 
             // get element prefix, namespace and name
-            if (ereg(':', $name)) {
+            if (preg_match('/:/', $name)) {
                 // get ns prefix
                 $prefix = substr($name, 0, strpos($name, ':')); 
                 // get ns
@@ -4990,7 +4990,7 @@
 	*/
 	function end_element($parser, $name){ 
 		// unset schema status
-		if (/*ereg('types$', $name) ||*/ ereg('schema$', $name)) {
+		if (/*ereg('types$', $name) ||*/ preg_match('/schema$/', $name)) {
 			$this->status = "";
             $this->appendDebug($this->currentSchema->getDebug());
             $this->currentSchema->clearDebug();
@@ -6587,7 +6587,7 @@
 			$key_localpart = $this->getLocalPart($key);
 			// if ns declarations, add to class level array of valid namespaces
             if($key_prefix == 'xmlns'){
-				if(ereg('^http://www.w3.org/[0-9]{4}/XMLSchema$',$value)){
+				if(preg_match('/^http:\/\/www.w3.org\/[0-9]{4}\/XMLSchema$/',$value)){
 					$this->XMLSchemaVersion = $value;
 					$this->namespaces['xsd'] = $this->XMLSchemaVersion;
 					$this->namespaces['xsi'] = $this->XMLSchemaVersion.'-instance';
@@ -6615,7 +6615,7 @@
 				}
 			} elseif($key_localpart == 'arrayType'){
 				$this->message[$pos]['type'] = 'array';
-				/* do arrayType ereg here
+				/* do arrayType preg here
 				[1]    arrayTypeValue    ::=    atype asize
 				[2]    atype    ::=    QName rank*
 				[3]    rank    ::=    '[' (',')* ']'
@@ -6624,7 +6624,7 @@
 				[6]    nextDimension    ::=    Digit+ ','
 				*/
 				$expr = '([A-Za-z0-9_]+):([A-Za-z]+[A-Za-z0-9_]+)\[([0-9]+),?([0-9]*)\]';
-				if(ereg($expr,$value,$regs)){
+				if(preg_match("/$expr/",$value,$regs)){
 					$this->message[$pos]['typePrefix'] = $regs[1];
 					$this->message[$pos]['arrayTypePrefix'] = $regs[1];
 	                if (isset($this->namespaces[$regs[1]])) {
@@ -7422,7 +7422,7 @@
 		// detect transport
 		switch(true){
 			// http(s)
-			case ereg('^http',$this->endpoint):
+			case preg_match('/^http/',$this->endpoint):
 				$this->debug('transporting via HTTP');
 				if($this->persistentConnection == true && is_object($this->persistentConnection)){
 					$http =& $this->persistentConnection;
@@ -7444,10 +7444,10 @@
 					$http->setEncoding($this->http_encoding);
 				}
 				$this->debug('sending message, length='.strlen($msg));
-				if(ereg('^http:',$this->endpoint)){
+				if(preg_match('/^http:/',$this->endpoint)){
 				//if(strpos($this->endpoint,'http:')){
 					$this->responseData = $http->send($msg,$timeout,$response_timeout,$this->cookies);
-				} elseif(ereg('^https',$this->endpoint)){
+				} elseif(preg_match('/^https/',$this->endpoint)){
 				//} elseif(strpos($this->endpoint,'https:')){
 					//if(phpversion() == '4.3.0-dev'){
 						//$response = $http->send($msg,$timeout,$response_timeout);
@@ -7506,7 +7506,7 @@
 		if (strpos($headers['content-type'], '=')) {
 			$enc = str_replace('"', '', substr(strstr($headers["content-type"], '='), 1));
 			$this->debug('Got response encoding: ' . $enc);
-			if(eregi('^(ISO-8859-1|US-ASCII|UTF-8)$',$enc)){
+			if(preg_match('/^(ISO-8859-1|US-ASCII|UTF-8)$/i',$enc)){
 				$this->xml_encoding = strtoupper($enc);
 			} else {
 				$this->xml_encoding = 'US-ASCII';
diff -ruN fusionforge-4.8.3-original/www/survey/admin/show_results_csv.php fusionforge-4.8.3-ereg_preg/www/survey/admin/show_results_csv.php
--- fusionforge-4.8.3-original/www/survey/admin/show_results_csv.php	2009-06-07 10:48:52.000000000 -0700
+++ fusionforge-4.8.3-ereg_preg/www/survey/admin/show_results_csv.php	2010-05-06 03:57:42.299088944 -0700
@@ -37,7 +37,7 @@
 }
 
 function strip_commas($string) {
-	return ereg_replace(",","",$string);
+	return preg_replace("/,/","",$string);
 }
 
 /*
