diff -ruN fusionforge-4.8.3-original/www/project/request.php fusionforge-4.8.3-misc_fixes/www/project/request.php
--- fusionforge-4.8.3-original/www/project/request.php	2010-01-16 06:27:30.000000000 -0800
+++ fusionforge-4.8.3-misc_fixes/www/project/request.php	2010-05-13 17:38:56.475699662 -0700
@@ -48,7 +48,7 @@
 	if (!$gjr->create($usr->getId(),$comments)) {
 		exit_error('Error',$gjr->getErrorMessage());
 	} else {
-		$feedback .= _('Your request has been submitted.');
+		$feedback = _('Your request has been submitted.');
 	}
 }
 
