--- bin/svn_update.pl.orig	2006-10-30 14:31:21.000000000 +0100
+++ bin/svn_update.pl	2006-10-30 14:36:17.000000000 +0100
@@ -58,13 +58,13 @@
 		# First create the repository
 		# Unix right will lock access to all users not in the
 		# group including ViewCVS when anoncvs is not enabled
-		mkdir $cvs_dir, 0775;
+		mkdir $svn_dir, 0775;
 		# Used fsfs backend because ViewCVS (apache) needs
 		# write permission with default backend
 		system("/usr/bin/svnadmin create --fs-type fsfs $svn_dir");
 		
 		# set group ownership, anonymous group user
-		system("chown -R $svn_uid:$svn_gid $svn_dir");
+		system("chown -R apache:apache $svn_dir");
 		system("chmod -R g+rw $svn_dir");
 		# s bit to have all owned by group
 		system("find $svn_dir -type d | xargs chmod g+s");
@@ -73,11 +73,11 @@
 	}
 
 	# Right management
-	if ($group_exists && $use_scm && $status eq 'A'){
-	    if ($enable_anonscm) {
-		chmod 02775, "$svn_dir";
-	    } else {
-		chmod 02070, "$svn_dir";
-	    }
-	}
+	#if ($group_exists && $use_scm && $status eq 'A'){
+	#    if ($enable_anonscm) {
+	#	chmod 02775, "$svn_dir";
+	#    } else {
+	#	chmod 02070, "$svn_dir";
+	#    }
+	#}
 }
