--- /usr/src/packages/SPECS/libnss-pgsql.spec	2021-03-22 10:58:15.126140222 +0100
+++ /usr/src/packages/SPECS/libnss-pgsql.spec	2021-03-22 10:58:45.866139600 +0100
@@ -10,7 +10,7 @@
 Patch1:         libnss-pgsql-1.5.0-beta-exit-in-library.patch
 
 BuildRequires:  gcc
-BuildRequires:  libpq-devel, xmlto
+BuildRequires:  postgresql-devel, xmlto
 
 
 %description
@@ -25,7 +25,7 @@
 
 
 %build
-%configure --disable-static
+CFLAGS="-I/usr/include/pgsql" %configure --disable-static
 make %{?_smp_mflags}
 
 
@@ -36,9 +36,6 @@
 find $RPM_BUILD_ROOT -name '*.la' -exec rm -f {} ';'
 
 
-%ldconfig_scriptlets
-
-
 %files
 %exclude %{_libdir}/*.so
 %doc AUTHORS COPYING ChangeLog README doc/nss-pgsql.html doc/caution.png
