--- phpDocumentor.ini.orig	2010-02-25 11:48:58.918048044 +0100
+++ phpDocumentor.ini	2010-02-25 11:49:19.668076615 +0100
@@ -22,7 +22,7 @@
 
 ;; set max memory usage size to be very high, to avoid it crashing it the middle of its run
 ;; due to using a boatload of memory
-;;memory_limit = 512M
+memory_limit = 1536M
 
 [_phpDocumentor_phpfile_exts]
 php
@@ -30,6 +30,7 @@
 php4
 phtml
 inc
+class
 
 ;; deprecated in 1.2
 ;; this list is informational only - the following tags will be parsed as
