--- FusionForge.php.renamed	2010-04-29 23:18:01.000000000 +0200
+++ FusionForge.php	2010-04-29 23:05:54.000000000 +0200
@@ -34,18 +34,19 @@
 		parent::setupSkinUserCss( $out );
 
 		// Append to the default screen common & print styles...
-		$out->addStyle( 'fusionforge/main.css', 'screen' );
+		$out->addStyle( 'monobook/main.css', 'screen' );
+		$out->addStyle( 'fusionforge/fusionforge.css', 'screen' );
 		if( $wgHandheldStyle ) {
 			// Currently in testing... try 'chick/main.css'
 			$out->addStyle( $wgHandheldStyle, 'handheld' );
 		}
 
-		$out->addStyle( 'fusionforge/IE50Fixes.css', 'screen', 'lt IE 5.5000' );
-		$out->addStyle( 'fusionforge/IE55Fixes.css', 'screen', 'IE 5.5000' );
-		$out->addStyle( 'fusionforge/IE60Fixes.css', 'screen', 'IE 6' );
-		$out->addStyle( 'fusionforge/IE70Fixes.css', 'screen', 'IE 7' );
+		$out->addStyle( 'monobook/IE50Fixes.css', 'screen', 'lt IE 5.5000' );
+		$out->addStyle( 'monobook/IE55Fixes.css', 'screen', 'IE 5.5000' );
+		$out->addStyle( 'monobook/IE60Fixes.css', 'screen', 'IE 6' );
+		$out->addStyle( 'monobook/IE70Fixes.css', 'screen', 'IE 7' );
 
-		$out->addStyle( 'fusionforge/rtl.css', 'screen', '', 'rtl' );
+		$out->addStyle( 'monobook/rtl.css', 'screen', '', 'rtl' );
 	}
 }
 
@@ -80,6 +81,10 @@
 		<meta http-equiv="Content-Type" content="<?php $this->text('mimetype') ?>; charset=<?php $this->text('charset') ?>" />
 		<?php $this->html('headlinks') ?>
 		<title><?php $this->text('pagetitle') ?></title>
+        	<!-- FUSIONFORGE Stylesheet BEGIN -->
+		<?php if (!$GLOBALS['sys_use_mwframe']){ $GLOBALS['HTML']->headerLink(); } ?>
+		<?php if (!$GLOBALS['sys_use_mwframe']){ $GLOBALS['HTML']->headerCSS(); } ?>
+        	<!-- FUSIONFORGE Stylesheet END -->
 		<?php $this->html('csslinks') ?>
 
 		<!--[if lt IE 7]><script type="<?php $this->text('jsmimetype') ?>" src="<?php $this->text('stylepath') ?>/common/IEFixes.js?<?php echo $GLOBALS['wgStyleVersion'] ?>"></script>
@@ -107,9 +112,22 @@
 <?php	}
 		if($this->data['trackbackhtml']) print $this->data['trackbackhtml']; ?>
 	</head>
-<body<?php if($this->data['body_ondblclick']) { ?> ondblclick="<?php $this->text('body_ondblclick') ?>"<?php } ?>
+<body id="mydoc" <?php if($this->data['body_ondblclick']) { ?> ondblclick="<?php $this->text('body_ondblclick') ?>"<?php } ?>
 <?php if($this->data['body_onload']) { ?> onload="<?php $this->text('body_onload') ?>"<?php } ?>
  class="mediawiki <?php $this->text('dir') ?> <?php $this->text('pageclass') ?> <?php $this->text('skinnameclass') ?>">
+        <!-- FUSIONFORGE BodyHeader BEGIN -->
+        <?php
+	if (!$GLOBALS['sys_use_mwframe']){
+        	$project=group_get_object_by_name($GLOBALS['fusionforgeproject']);
+        	if ($project) {
+			$GLOBALS['group_id']=$project->getID();
+                	$params['group']=$GLOBALS['group_id'];
+                	$params['toptab']='mediawiki';
+                	$GLOBALS['HTML']->bodyHeader($params);
+        	}
+	}
+        ?>
+        <!-- FUSIONFORGE BodyHeader END -->
 	<div id="globalWrapper">
 		<div id="column-content">
 	<div id="content">
@@ -242,7 +260,12 @@
 
 -->
 <?php endif; ?>
+        <!-- FUSIONFORGE Footer BEGIN -->
+<?php	if (!$GLOBALS['sys_use_mwframe']){
+		$GLOBALS['HTML']->footer($params);
+	} else { ?>
 </body></html>
+<?php	} ?>
 <?php
 	wfRestoreWarnings();
 	} // end of execute() method
