diff -Nur -x '*.orig' -x '*.rej' gforge-3.0/etc/local.inc mezzanine_patched_gforge-3.0/etc/local.inc
--- gforge-3.0/etc/local.inc	Fri Jun 20 07:42:09 2003
+++ mezzanine_patched_gforge-3.0/etc/local.inc	Thu Aug 14 20:13:10 2003
@@ -27,13 +27,13 @@
 // The default GForge domain
 // this is used where ever the "naked" form of the GForge domain
 // might be used.  E.g., "mailto:admin@gforge.net"
-$sys_default_domain = 'gforge.company.com';
-$sys_fallback_domain = 'gforge2.company.com';
-$sys_internal_network = '192.168.1.';
+$sys_default_domain = 'localhost';
+$sys_fallback_domain = 'localhost';
+$sys_internal_network = '127.0.0.';
 
 // Machine that hosts CVS
 $sys_cvs_single_host=true;
-$sys_cvs_host = 'cvs.company.com';
+$sys_cvs_host = 'localhost';
 
 // Force the use of a single cvs host instead of cvs.project.domain.com
 // Set to 1 to use cvs.domain.com:/cvsroot/project for all projects
@@ -41,17 +41,17 @@
 $sys_cvs_single_host = 1;
 
 // Machine used for downloading sources/packages
-$sys_download_host = "download.company.com";
+$sys_download_host = "localhost";
 
 // Machine used for uploading sources/packages
-$sys_upload_host = "upload.company.com";
+$sys_upload_host = "localhost";
 
 // Machine(s) that host users' shell accounts
 //  N.B. to the SourceForge Crew: What's the difference between the user
 // host and the shell host?  They are clearly two different hostnames
 // in the source code, but they seem to serve the same purpose..?
-$sys_shell_host = "shell.company.com";
-$sys_users_host = "users.company.com";
+$sys_shell_host = "localhost";
+$sys_users_host = "localhost";
 $homedir_prefix='/home';
 $groupdir_prefix='/home/groups';
 
@@ -60,18 +60,18 @@
 // machine that the SourceForge code is running on because all of the
 // PHP makes reference to these documents in terms of relative paths that
 // are part of the GForge code tree.
-$sys_docs_host = "sfdocs.company.com";
+$sys_docs_host = "localhost";
 
 // Machine that hosts the GForge mailing lists (This could also be
 // the mail host if you have enough horsepower & bandwidth)
-$sys_lists_host = "lists.company.com";
+$sys_lists_host = "localhost";
 
 // Domain Name Servers
 //    N.B.: Use terminated FQDNs here (with the final ".") so the resolver
 //    doesn't attempt to recurse in the case of a slightly broken DNS
 //    configuration
-$sys_dns1_host = "ns1.company.com";
-$sys_dns2_host = "ns2.company.com";
+$sys_dns1_host = "localhost";
+$sys_dns2_host = "localhost";
 
 //Databases, html/php/other paths
 //server to use for updates and reads
@@ -83,8 +83,8 @@
 $sys_dbreadhost='localhost';
 $sys_dbreaddb='stats';
 $sys_dbname="alexandria";
-$sys_dbuser="www";
-$sys_dbpasswd="";
+$sys_dbuser="gforge";
+$sys_dbpasswd="DBPASSHERE";
 // You can also specify a database port if you're using something other than 5432
 //$sys_dbport="4242";
 $sys_server="pgsql";
@@ -95,7 +95,7 @@
 //
 // disable ldap use altogether
 $sys_use_ldap=0;
-$sys_ldap_host="ldap.company.com";
+$sys_ldap_host="localhost";
 $sys_ldap_port=389;
 // this is dn under which all information stored
 $sys_ldap_base_dn="dc=company,dc=com";
@@ -113,7 +113,7 @@
 //
 $sys_use_jabber=0;
 	//messages from the system will be sent to this address
-$sys_jabber_server='jabber.company.com';
+$sys_jabber_server='localhost';
 $sys_jabber_port='5222';
 	// messages sent to jabber accounts will come from this user
 	// It is similar to the "From: noreply@gforge.org" used in emails
@@ -128,7 +128,7 @@
 //	Your php.ini file may have to be modified to allow writing outside
 //	the webserver's directory
 //
-$sys_upload_dir='/path/to/uploads/';
+$sys_upload_dir='/usr/share/gforge/www/incoming';
 
 // Where the GForge files are placed
 // *** IMPORTANT: sys_urlroot *MUST* be an ABSOLUTE FILEYSTEM PATH NAME
@@ -136,16 +136,16 @@
 //             installation.  If you use ANY form of relative path
 //             you will break the html_image function in include/html.php
 //
-$sys_urlroot="/path/to/gforge/www/";
+$sys_urlroot="/usr/share/gforge/www/";
 
 // Name of the system as a whole (needed by various utils and titles)
-$sys_name="prodigy";
+$sys_name="localhost";
 
 // session cookie settings
 //
 //	IMPORTANT - YOU MUST CHANGE "foobar" to a long, random number
 //
-$sys_session_key        = "foobar";
+$sys_session_key        = "RANDOMIDHERE";
 $sys_session_expire     = 60 * 60 * 24 * 7;
 
 // Require that user give unique (not yet existent in db) email upon
