--- cronjobs/tarballs.php.orig	2006-10-30 14:20:12.000000000 +0100
+++ cronjobs/tarballs.php	2006-10-30 14:20:54.000000000 +0100
@@ -6,7 +6,7 @@
 // For portability, if hot-backup.py path vary from one distro to another add some path
 // in the next line
 putenv("PATH=/usr/lib/subversion:".getenv('PATH'));
-$BACKUPPROG='hot-backup.py';
+$BACKUPPROG='/usr/bin/python /usr/share/doc/subversion-1.3.2/tools/backup/hot-backup.py';
 
 require ('squal_pre.php');
 require ('common/include/cron_utils.php');
