diff -ru etc.orig/plugins/scmsvn/config.php etc/plugins/scmsvn/config.php
--- etc.orig/plugins/scmsvn/config.php	2006-10-30 14:57:10.000000000 +0100
+++ etc/plugins/scmsvn/config.php	2006-10-30 14:57:45.000000000 +0100
@@ -1,7 +1,7 @@
 <?php
  
-//$default_svn_server = $GLOBALS['sys_default_domain'] ;
-$default_svn_server = "svn." . $GLOBALS['sys_default_domain'] ;
+$default_svn_server = $GLOBALS['sys_default_domain'] ;
+//$default_svn_server = "svn." . $GLOBALS['sys_default_domain'] ;
 //$default_svn_server = $GLOBALS['sys_scm_host'];
 $use_ssh = false;
 $use_dav = true;
diff -ru etc.orig/plugins/scmsvn/config.pl etc/plugins/scmsvn/config.pl
--- etc.orig/plugins/scmsvn/config.pl	2006-10-30 14:57:15.000000000 +0100
+++ etc/plugins/scmsvn/config.pl	2006-10-30 14:57:55.000000000 +0100
@@ -1,5 +1,5 @@
-$this_server = $sys_scm_host ;
-# $this_server = $sys_default_domain ;
+#$this_server = $sys_scm_host ;
+$this_server = $sys_default_domain ;
 # $this_server = "svn." . $sys_default_domain ;
 
 1 ;
