--- FusionForge.php.renamed	2010-02-24 20:26:14.000000000 +0100
+++ FusionForge.php	2010-02-24 20:26:14.000000000 +0100
@@ -80,6 +80,10 @@
 		<meta http-equiv="Content-Type" content="<?php $this->text('mimetype') ?>; charset=<?php $this->text('charset') ?>" />
 		<?php $this->html('headlinks') ?>
 		<title><?php $this->text('pagetitle') ?></title>
+        	<!-- FUSIONFORGE Stylesheet BEGIN -->
+		<?php if (!$GLOBALS['sys_use_mwframe']){ $GLOBALS['HTML']->headerLink(); } ?>
+		<?php if (!$GLOBALS['sys_use_mwframe']){ $GLOBALS['HTML']->headerCSS(); } ?>
+        	<!-- FUSIONFORGE Stylesheet END -->
 		<?php $this->html('csslinks') ?>
 
 		<!--[if lt IE 7]><script type="<?php $this->text('jsmimetype') ?>" src="<?php $this->text('stylepath') ?>/common/IEFixes.js?<?php echo $GLOBALS['wgStyleVersion'] ?>"></script>
@@ -110,6 +114,19 @@
 <body<?php if($this->data['body_ondblclick']) { ?> ondblclick="<?php $this->text('body_ondblclick') ?>"<?php } ?>
 <?php if($this->data['body_onload']) { ?> onload="<?php $this->text('body_onload') ?>"<?php } ?>
  class="mediawiki <?php $this->text('dir') ?> <?php $this->text('pageclass') ?> <?php $this->text('skinnameclass') ?>">
+        <!-- FUSIONFORGE BodyHeader BEGIN -->
+        <?php
+	if (!$GLOBALS['sys_use_mwframe']){
+        	$project=group_get_object_by_name($GLOBALS['fusionforgeproject']);
+        	if ($project) {
+			$GLOBALS['group_id']=$project->getID();
+                	$params['group']=$GLOBALS['group_id'];
+                	$params['toptab']='mediawiki';
+                	$GLOBALS['HTML']->bodyHeader($params);
+        	}
+	}
+        ?>
+        <!-- FUSIONFORGE BodyHeader END -->
 	<div id="globalWrapper">
 		<div id="column-content">
 	<div id="content">
@@ -242,7 +259,13 @@
 
 -->
 <?php endif; ?>
+        <!-- FUSIONFORGE Footer BEGIN -->
+<?php	if (!$GLOBALS['sys_use_mwframe']){
+		$GLOBALS['HTML']->footer($params);
+	} else { ?>
 </body></html>
+<?php	} ?>
+        <!-- FUSIONFORGE Footer END -->
 <?php
 	wfRestoreWarnings();
 	} // end of execute() method
