diff -Nur -x '*.orig' -x '*.rej' gforge-3.0pre9/db/gforge3.sql mezzanine_patched_gforge-3.0pre9/db/gforge3.sql
--- gforge-3.0pre9/db/gforge3.sql	Wed Jan 15 19:16:45 2003
+++ mezzanine_patched_gforge-3.0pre9/db/gforge3.sql	Fri Jan 24 16:05:09 2003
@@ -4035,7 +4035,7 @@
 SELECT setval ('"user_metric0_pk_seq"', 1, false);
 
 
-SELECT setval ('"users_pk_seq"', 101, true);
+SELECT setval ('"users_pk_seq"', 102, true);
 
 
 SELECT setval ('"unix_uid_seq"', 1, false);
@@ -4124,3 +4124,14 @@
 
 SELECT setval ('"user_plugin_pk_seq"', 1, false);
 
+INSERT INTO "users" ("user_id","user_name","email","user_pw","realname","status","shell","unix_pw","unix_status","unix_uid","unix_box","add_date","confirm_hash","mail_siteupdates","mail_va","authorized_keys","email_new","people_view_skills","people_resume","timezone","language","block_ratings","jabber_address","jabber_only") VALUES (101,'siteadmin','gforge@localhost','','GForge Site Admin','A','/bin/bash','','N',0,'shell1',1043359556,'',0,0,NULL,NULL,0,'','GMT',1,0,'',0);
+
+INSERT INTO user_group (user_id, group_id, admin_flags) VALUES (101, 1, 'A');
+INSERT INTO user_group (user_id, group_id, admin_flags) VALUES (101, 2, 'A');
+INSERT INTO user_group (user_id, group_id, admin_flags) VALUES (101, 3, 'A');
+INSERT INTO user_group (user_id, group_id, admin_flags) VALUES (101, 4, 'A');
+
+UPDATE groups SET homepage = '/admin/' WHERE group_id = 1;
+UPDATE groups SET homepage = '/news/' WHERE group_id = 2;
+UPDATE groups SET homepage = '/stats/' WHERE group_id = 3;
+UPDATE groups SET homepage = '/peerrating/' WHERE group_id = 4;
