diff -Nur -x '*.orig' -x '*.rej' gforge-3.0pre9/cronjobs/calculate_user_metric.php mezzanine_patched_gforge-3.0pre9/cronjobs/calculate_user_metric.php
--- gforge-3.0pre9/cronjobs/calculate_user_metric.php	Wed Jan 15 19:16:44 2003
+++ mezzanine_patched_gforge-3.0pre9/cronjobs/calculate_user_metric.php	Wed Jan 22 17:22:14 2003
@@ -1,4 +1,4 @@
-#! /usr/bin/php4 -f
+#! /usr/bin/php -f
 <?php
 /**
   *
diff -Nur -x '*.orig' -x '*.rej' gforge-3.0pre9/cronjobs/check_stale_tracker_items.php mezzanine_patched_gforge-3.0pre9/cronjobs/check_stale_tracker_items.php
--- gforge-3.0pre9/cronjobs/check_stale_tracker_items.php	Wed Jan 15 19:16:44 2003
+++ mezzanine_patched_gforge-3.0pre9/cronjobs/check_stale_tracker_items.php	Wed Jan 22 17:22:14 2003
@@ -1,4 +1,4 @@
-#! /usr/bin/php4 -f
+#! /usr/bin/php -f
 <?php
 /**
  * check_stale_tracker_items.php - Check for stale tracker items.
diff -Nur -x '*.orig' -x '*.rej' gforge-3.0pre9/cronjobs/db_stats_agg.php mezzanine_patched_gforge-3.0pre9/cronjobs/db_stats_agg.php
--- gforge-3.0pre9/cronjobs/db_stats_agg.php	Wed Jan 15 19:16:44 2003
+++ mezzanine_patched_gforge-3.0pre9/cronjobs/db_stats_agg.php	Wed Jan 22 17:22:14 2003
@@ -1,4 +1,4 @@
-#! /usr/bin/php4 -f
+#! /usr/bin/php -f
 <?php
 /**
  * GForge
diff -Nur -x '*.orig' -x '*.rej' gforge-3.0pre9/cronjobs/db_trove_maint.php mezzanine_patched_gforge-3.0pre9/cronjobs/db_trove_maint.php
--- gforge-3.0pre9/cronjobs/db_trove_maint.php	Wed Jan 15 19:16:44 2003
+++ mezzanine_patched_gforge-3.0pre9/cronjobs/db_trove_maint.php	Wed Jan 22 17:22:14 2003
@@ -1,4 +1,4 @@
-#! /usr/bin/php4 -f
+#! /usr/bin/php -f
 <?php
 /**
   *
diff -Nur -x '*.orig' -x '*.rej' gforge-3.0pre9/cronjobs/massmail.php mezzanine_patched_gforge-3.0pre9/cronjobs/massmail.php
--- gforge-3.0pre9/cronjobs/massmail.php	Wed Jan 15 19:16:44 2003
+++ mezzanine_patched_gforge-3.0pre9/cronjobs/massmail.php	Wed Jan 22 17:22:14 2003
@@ -1,4 +1,4 @@
-#! /usr/bin/php4 -f
+#! /usr/bin/php -f
 <?php
 /**
   *
diff -Nur -x '*.orig' -x '*.rej' gforge-3.0pre9/cronjobs/project_cleanup.php mezzanine_patched_gforge-3.0pre9/cronjobs/project_cleanup.php
--- gforge-3.0pre9/cronjobs/project_cleanup.php	Wed Jan 15 19:16:44 2003
+++ mezzanine_patched_gforge-3.0pre9/cronjobs/project_cleanup.php	Wed Jan 22 17:22:14 2003
@@ -1,4 +1,4 @@
-#! /usr/bin/php4 -f
+#! /usr/bin/php -f
 <?php
 /**
   *
diff -Nur -x '*.orig' -x '*.rej' gforge-3.0pre9/cronjobs/project_weekly_metric-backfill.php mezzanine_patched_gforge-3.0pre9/cronjobs/project_weekly_metric-backfill.php
--- gforge-3.0pre9/cronjobs/project_weekly_metric-backfill.php	Wed Jan 15 19:16:44 2003
+++ mezzanine_patched_gforge-3.0pre9/cronjobs/project_weekly_metric-backfill.php	Wed Jan 22 17:22:14 2003
@@ -1,4 +1,4 @@
-#! /usr/bin/php4 -f
+#! /usr/bin/php -f
 <?php
 /**
   *
diff -Nur -x '*.orig' -x '*.rej' gforge-3.0pre9/cronjobs/project_weekly_metric.php mezzanine_patched_gforge-3.0pre9/cronjobs/project_weekly_metric.php
--- gforge-3.0pre9/cronjobs/project_weekly_metric.php	Wed Jan 15 19:16:44 2003
+++ mezzanine_patched_gforge-3.0pre9/cronjobs/project_weekly_metric.php	Wed Jan 22 17:22:14 2003
@@ -1,4 +1,4 @@
-#! /usr/bin/php4 -f
+#! /usr/bin/php -f
 <?php
 /**
   *
diff -Nur -x '*.orig' -x '*.rej' gforge-3.0pre9/cronjobs/rating_stats.php mezzanine_patched_gforge-3.0pre9/cronjobs/rating_stats.php
--- gforge-3.0pre9/cronjobs/rating_stats.php	Wed Jan 15 19:16:44 2003
+++ mezzanine_patched_gforge-3.0pre9/cronjobs/rating_stats.php	Wed Jan 22 17:22:14 2003
@@ -1,4 +1,4 @@
-#! /usr/bin/php4 -f
+#! /usr/bin/php -f
 <?php
 /**
   *
diff -Nur -x '*.orig' -x '*.rej' gforge-3.0pre9/cronjobs/rotate_activity.php mezzanine_patched_gforge-3.0pre9/cronjobs/rotate_activity.php
--- gforge-3.0pre9/cronjobs/rotate_activity.php	Wed Jan 15 19:16:44 2003
+++ mezzanine_patched_gforge-3.0pre9/cronjobs/rotate_activity.php	Wed Jan 22 17:22:14 2003
@@ -1,4 +1,4 @@
-#! /usr/bin/php4 -f
+#! /usr/bin/php -f
 <?php
 /**
   *
diff -Nur -x '*.orig' -x '*.rej' gforge-3.0pre9/cronjobs/site_stats.php mezzanine_patched_gforge-3.0pre9/cronjobs/site_stats.php
--- gforge-3.0pre9/cronjobs/site_stats.php	Wed Jan 15 19:16:44 2003
+++ mezzanine_patched_gforge-3.0pre9/cronjobs/site_stats.php	Wed Jan 22 17:22:14 2003
@@ -1,4 +1,4 @@
-#! /usr/bin/php4 -f
+#! /usr/bin/php -f
 <?php
 /**
   *
diff -Nur -x '*.orig' -x '*.rej' gforge-3.0pre9/cronjobs/stats_projects-backfill.php mezzanine_patched_gforge-3.0pre9/cronjobs/stats_projects-backfill.php
--- gforge-3.0pre9/cronjobs/stats_projects-backfill.php	Wed Jan 15 19:16:44 2003
+++ mezzanine_patched_gforge-3.0pre9/cronjobs/stats_projects-backfill.php	Wed Jan 22 17:22:14 2003
@@ -1,4 +1,4 @@
-#! /usr/bin/php4 -f
+#! /usr/bin/php -f
 <?php
 /**
   *
diff -Nur -x '*.orig' -x '*.rej' gforge-3.0pre9/cronjobs/vacuum.php mezzanine_patched_gforge-3.0pre9/cronjobs/vacuum.php
--- gforge-3.0pre9/cronjobs/vacuum.php	Wed Jan 15 19:16:44 2003
+++ mezzanine_patched_gforge-3.0pre9/cronjobs/vacuum.php	Wed Jan 22 17:22:14 2003
@@ -1,4 +1,4 @@
-#! /usr/bin/php4 -f
+#! /usr/bin/php -f
 <?php
 
 /**
diff -Nur -x '*.orig' -x '*.rej' gforge-3.0pre9/db/20021213_doc_data-migrate.php mezzanine_patched_gforge-3.0pre9/db/20021213_doc_data-migrate.php
--- gforge-3.0pre9/db/20021213_doc_data-migrate.php	Wed Jan 15 19:16:45 2003
+++ mezzanine_patched_gforge-3.0pre9/db/20021213_doc_data-migrate.php	Wed Jan 22 17:22:14 2003
@@ -1,4 +1,4 @@
-#! /usr/bin/php4 -f
+#! /usr/bin/php -f
 <?php
 
 /**
diff -Nur -x '*.orig' -x '*.rej' gforge-3.0pre9/db/SQL_migrate-2.5-to-2.6/artifact-convert-files.php mezzanine_patched_gforge-3.0pre9/db/SQL_migrate-2.5-to-2.6/artifact-convert-files.php
--- gforge-3.0pre9/db/SQL_migrate-2.5-to-2.6/artifact-convert-files.php	Wed Jan 15 19:16:45 2003
+++ mezzanine_patched_gforge-3.0pre9/db/SQL_migrate-2.5-to-2.6/artifact-convert-files.php	Wed Jan 22 17:22:20 2003
@@ -1,4 +1,4 @@
-#!/usr/local/bin/php
+#!/usr/bin/php
 <?php
 
 require ('squal_pre.php');
