Index: www/project/admin/index.php
===================================================================
RCS file: /cvsroot/gforge/gforge/www/project/admin/index.php,v
retrieving revision 1.11
diff -u -r1.11 index.php
--- www/project/admin/index.php	10 Jan 2003 14:44:25 -0000	1.11
+++ www/project/admin/index.php	10 Jan 2003 16:47:31 -0000
@@ -155,12 +155,36 @@
 	// After adding user, we go to the permission page for one
 ?>
 	<hr noshade="noshade size="1" />
-	<form action="userpermedit.php?group_id=<?php echo $group->getID(); ?>" method="post">
+	<form name="addform" action="userpermedit.php?group_id=<?php echo $group->getID(); ?>" method="post">
 	<input type="hidden" name="func" value="adduser" />
 	<table width="100%" border="0">
 	<tr><td><strong>Unix Name:</strong></td><td><input type="text" name="form_unix_name" size="10" value="" /></td></tr>
-	<tr><td colspan="2" align="center"><input type="submit" name="submit" value="Add User" /></td></tr></form>
-	</table>
+
+
+	<tr><td align="center" colspan="2"> 
+<?php
+	$db_users = db_query("SELECT users.realname , users.user_name ".
+	 	"FROM users WHERE user_name != 'None' ".
+ 		"ORDER BY  users.realname");
+ 
+         print '<select onchange="updateUser()" width="100" multiple="no" name="selected_users" size="10">';
+ 
+         while ($row_user=db_fetch_array($db_users)) {
+	 print '<option>'.$row_user['realname'].'&nbsp;&nbsp;('.$row_user['user_name'].')</option>';
+ }
+         print '</select>';
+ ?>
+        </td></tr>
+	<tr><td colspan="2" align="center"><input type="submit" name="submit" value="Add User" /></td></tr>	</table>
+	</form>
+ <script language="JavaScript">
+ function updateUser() {
+     var user = document.forms['addform'].selected_users.options[document.forms['addform'].selected_users.selectedIndex].text;
+     var start = user.indexOf('(');
+     var end = user.indexOf(')');
+     document.forms['addform'].form_unix_name.value = user.substring(start+1, end);
+ }
+ </script>
 
 	<hr noshade="noshade size="1" />
 	<div align="center">
